<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Statistical Inference of Associations between Microbial
    Communities And host phenoTypes • SIAMCAT</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Statistical Inference of Associations between Microbial
    Communities And host phenoTypes">
<meta property="og:description" content="Pipeline for Statistical Inference of Associations between
    Microbial Communities And host phenoTypes (SIAMCAT). A primary goal
    of analyzing microbiome data is to determine changes in community
    composition that are associated with environmental factors. In particular,
    linking human microbiome composition to host phenotypes such as diseases
    has become an area of intense research. For this, robust statistical
    modeling and biomarker extraction toolkits are crucially needed. SIAMCAT
    provides a full pipeline supporting data preprocessing, statistical
    association testing, statistical modeling (LASSO logistic regression)
    including tools for evaluation and interpretation of these models (such as
    cross validation, parameter selection, ROC analysis and diagnostic
    model plots).">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><script type="text/javascript">
  var _paq = _paq || [];
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//tr-denbi.embl.de/heimdall/";
    _paq.push(['setTrackerUrl', u+'p.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'p.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">SIAMCAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/SIAMCAT_vignette.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/SIAMCAT_vignette.html">Introduction to SIAMCAT</a>
    </li>
    <li>
      <a href="articles/SIAMCAT_read-in.html">SIAMCAT input formats</a>
    </li>
    <li>
      <a href="articles/SIAMCAT_holdout.html">Holdout testing with SIAMCAT</a>
    </li>
  </ul>
</li>
<li>
  <a href="reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://twitter.com/siamcat_dev">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/zellerlab/siamcat">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">
<div id="siamcat" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#siamcat" class="anchor"></a>SIAMCAT <img src="reference/figures/logo.png" align="right" width="120">
</h1></div>

<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p><code>SIAMCAT</code> is a pipeline for Statistical Inference of Associations between Microbial Communities And host phenoTypes. A primary goal of analyzing microbiome data is to determine changes in community composition that are associated with environmental factors. In particular, linking human microbiome composition to host phenotypes such as diseases has become an area of intense research. For this, robust statistical modeling and biomarker extraction toolkits are crucially needed. <code>SIAMCAT</code> provides a full pipeline supporting data preprocessing, statistical association testing, statistical modeling (LASSO logistic regression) including tools for evaluation and interpretation of these models (such as cross validation, parameter selection, ROC analysis and diagnostic model plots).</p>
<p><a href="https://microbiome-tools.embl.de"> <img src="reference/figures/embl_microbiome_tools_logo.png" align="right" width="200"></a></p>
<p><code>SIAMCAT</code> is developed in the <a href="https://www.embl.de/research/units/scb/zeller/index.html">Zeller group</a> and is part of the suite of computational microbiome analysis tools hosted at <a href="https://www.embl.org/">EMBL</a>.</p>
</div>
<div id="starting-with-siamcat" class="section level2">
<h2 class="hasAnchor">
<a href="#starting-with-siamcat" class="anchor"></a>Starting with SIAMCAT</h2>
<div id="installation" class="section level3">
<h3 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h3>
<p>In order to start with <code>SIAMCAT</code>, you need to install it from Bioconductor:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ns-load">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))
    <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"BiocManager"</span>)
BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"SIAMCAT"</span>, <span class="dt">version =</span> <span class="st">"3.8"</span>)</code></pre></div>
<p>Alternatively, you can install the current development version via <code>devtools</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">require</a></span>(<span class="st">"devtools"</span>)
devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="dt">repo =</span> <span class="st">'zellerlab/siamcat'</span>)</code></pre></div>
</div>
<div id="quick-start" class="section level3">
<h3 class="hasAnchor">
<a href="#quick-start" class="anchor"></a>Quick start</h3>
<p>There are a few manuals that will kick-start you and help you analyse your data with <code>SIAMCAT</code>. You can find links to those on the <a href="https://bioconductor.org/packages/release/bioc/html/SIAMCAT.html">Bioconductor website of SIAMCAT</a> or you can type into <code>R</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/browseVignettes">browseVignettes</a></span>(<span class="st">"SIAMCAT"</span>)
<span class="co"># Please Note:</span>
<span class="co"># `browseVignettes` only works if `SIAMCAT` has been installed via Bioconductor</span></code></pre></div>
</div>
</div>
<div id="contact" class="section level2">
<h2 class="hasAnchor">
<a href="#contact" class="anchor"></a>Contact</h2>
<p>If you run into any issue:</p>
<ul>
<li>create an <a href="https://github.com/zellerlab/siamcat/issues/new">issue in this repository</a> or</li>
<li>mail <a href="mailto:zeller@embl.de">Georg Zeller</a> or</li>
<li>ask at the <a href="https://groups.google.com/forum/#!forum/siamcat-users">SIAMCAT support group</a>
</li>
</ul>
<p>If you found <code>SIAMCAT</code> useful, please consider giving us <a href="https://www.surveymonkey.de/r/denbi-service?sc=hd-hub&amp;tool=siamcat">feedback</a>.</p>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>If you use <code>SIAMCAT</code>, please cite us by using</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/citation">citation</a></span>(<span class="st">"SIAMCAT"</span>)</code></pre></div>
<p>or by</p>
<blockquote>
<p>Zych K, Wirbel J, Essex M, Breuer K, Karcher N, Costea PI, Sunagawa S, Bork P, Zeller G (2018). <em>SIAMCAT: Statistical Inference of Associations between Microbial Communities And host phenoTypes</em>. doi: 10.18129/B9.bioc.SIAMCAT (URL: <a href="http://doi.org/10.18129/B9.bioc.SIAMCAT" class="uri">http://doi.org/10.18129/B9.bioc.SIAMCAT</a>), R package version 1.0.1, (URL: <a href="https://bioconductor.org/packages/SIAMCAT/" class="uri">https://bioconductor.org/packages/SIAMCAT/</a>).</p>
</blockquote>
</div>
<div id="examples-of-primary-package-output" class="section level2">
<h2 class="hasAnchor">
<a href="#examples-of-primary-package-output" class="anchor"></a>Examples of primary package output</h2>
<p>To give you a small preview about the primary package output, here are some example plots taking from the main <code>SIAMCAT</code> vignette.</p>
<p>In this vignette, we use an example dataset which is also included in the <code>SIAMCAT</code> package. The dataset is taken from the publication of <a href="http://europepmc.org/abstract/MED/25432777">Zeller et al</a>, which demonstrated the potential of microbial species in fecal samples to distinguish patients with colorectal cancer (CRC) from healthy controls.</p>
<div id="association-testing" class="section level3">
<h3 class="hasAnchor">
<a href="#association-testing" class="anchor"></a>Association testing</h3>
<p>The result of the <code>check.associations</code> function is an association plot. For significantly associated microbial features, the plot shows:</p>
<ul>
<li>the abundances of the features across the two different classes (CRC vs. controls)</li>
<li>the significance of the enrichment calculated by a Wilcoxon test (after multiple hypothesis testing correction)</li>
<li>the generalized fold change of each feature</li>
<li>the prevalence shift between the two classes, and</li>
<li>the Area Under the Receiver Operating Characteristics Curve (AU-ROC) as non-parametric effect size measure.</li>
</ul>
<p><img src="reference/figures/associations_plot.png" alt="Association testing"></p>
</div>
<div id="model-interpretation-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#model-interpretation-plot" class="anchor"></a>Model interpretation plot</h3>
<p>After statistical models have been trained to distinguish cancer cases from controls, the models can be investigated by the function <code>model.interpretation.plot</code>. The plots shows:</p>
<ul>
<li>the median relative feature weight for selected features (barplot on the left)</li>
<li>the robustness of features (i.e. in how many of the models the specific feature has been selected)</li>
<li>the distribution of selected features across samples (central heatmap)</li>
<li>which proportion of the weight of all different models are shown in the plot (boxplot on the right), and</li>
<li>distribution of metadata across samples (heatmap below).</li>
</ul>
<p><img src="reference/figures/interpretation_plot.png" alt="Model interpretation plot"></p>
</div>
</div>
<div id="where-siamcat-has-been-used-already" class="section level2">
<h2 class="hasAnchor">
<a href="#where-siamcat-has-been-used-already" class="anchor"></a>Where SIAMCAT has been used already</h2>
<p>Several publications already used <code>SIAMCAT</code> (or previous versions thereof).</p>
<ul>
<li>
<p><strong><a href="http://europepmc.org/abstract/MED/25432777">Potential of fecal microbiota for early-stage detection of colorectal cancer</a></strong><br><em>Zeller G, Tap J, Voigt AY, Sunagawa S, Kultima JR, Costea PI, Amiot A, Böhm J, Brunetti F, Habermann N, Hercog R, Koch M, Luciani A, Mende DR, Schneider MA, Schrotz-King P, Tournigand C, Tran Van Nhieu J, Yamada T, Zimmermann J, Benes V, Kloor M, Ulrich CM, von Knebel Doeberitz M, Sobhani I, Bork P</em><br>
Molecular Systems Biology, (<strong>2014</strong>) 10, 766</p>
<blockquote>
<p>Original Publication that inspired <code>SIAMCAT</code></p>
</blockquote>
</li>
<li>
<p><strong><a href="https://doi.org/10.1016/j.ebiom.2016.01.032">Gut Microbiota Linked to Sexual Preference and HIV Infection</a></strong><br><em>Noguera-Julian M, Rocafort M, Guillén Y, Rivera J, Casadellà M, Nowak P, Hildebrand F, Zeller G, Parera M, Bellido R, Rodríguez C,Carrillo J, Mothe B, Coll J, Bravo I, Estany C, Herrero C, Saz J, Sirera G, Torrela A, Navarro J, Crespo M, Brander C, Negredo E, Blanco J, Guarner F, Calle ML, Bork P, Sönnerborgd A, Clotet B, Paredes R</em><br>
EBioMedicine 5 (<strong>2016</strong>) 135-146</p>
<blockquote>
<p>See Figure 5</p>
</blockquote>
</li>
<li>
<p><strong><a href="https://elifesciences.org/articles/42693">Extensive transmission of microbes along the gastrointestinal tract</a></strong><br><em>Schmidt TSB, Hayward MR, Coelho LP, Li SS, Costea PI, Voigt AY, Wirbel J, Maistrenko OM, Alves RJC, Bergsten E, de Beaufort C, Sobhani I, Heintz-Buschart A, Sunagawa S, Zeller G, Wilmes P, Bork P</em><br>
eLife, (<strong>2019</strong>) 8:e42693</p>
<blockquote>
<p>See Figure 3 - figure supplement 1</p>
</blockquote>
</li>
<li>
<p><strong><a href="https://www.nature.com/articles/s41591-019-0406-6">Meta-analysis of fecal metagenomes reveals global microbial signatures that are specific for colorectal cancer</a></strong><br><em>Wirbel J, Pyl PT, Kartal E, Zych K, Kashani A, Milanese A, Fleck JS, Voigt AY, Palleja A, Ponnudurai R, Sunagawa S, Coelho LP, Schrotz-King P, Vogtmann E, Habermann N, Niméus E, Thomas AM, Manghi P, Gandini S, Serrano D, Mizutani S, Shiroma H, Shiba S, Shibata T, Yachida S, Yamada T, Waldron L, Naccarati A, Segata N, Sinha R, Ulrich CM, Brenner H, Arumugam M, Bork P, Zeller G</em><br>
Nature Medicine, (<strong>2019</strong>) [Epub ahead of print]</p>
<blockquote>
<p>In this publication, <code>SIAMCAT</code> is used extensively for holdout testing</p>
</blockquote>
</li>
</ul>
<p>If you used <code>SIAMCAT</code> in your publication, <a href="mailto:zeller@embl.de">let us know</a>!</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/SIAMCAT">https://​www.bioconductor.org/​packages/​SIAMCAT</a>
</li>
<li>Check out the the source code at <br><a href="https://github.com/zellerlab/siamcat">https://​github.com/​zellerlab/​siamcat</a>
</li>
<li>SIAMCAT is part of the EMBL microbiome tools. Find out more at <br><a href="https://microbiome-tools.embl.de">https://​microbiome-tools.embl.de</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing SIAMCAT</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Konrad Zych <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-7426-0516" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" alt="ORCID" height="16"></a> </li>
<li>Jakob Wirbel <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-4073-3562" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" alt="ORCID" height="16"></a> </li>
<li>Georg Zeller <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-1429-7485" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" alt="ORCID" height="16"></a> </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.com/zellerlab/siamcat"><img src="https://travis-ci.com/zellerlab/siamcat.svg?branch=master" alt="Build Status"></a></li>
</ul>
</div>
</div>
</div>

      <footer><div class="copyright">
  <p>Developed by Konrad Zych, Jakob Wirbel, Georg Zeller.</p>
</div>

<div class="privacy">
  <p><strong>Privacy:</strong> Usage of this site follows
      <a href="https://www.embl.de/aboutus/privacy_policy/index.html">EMBL’s Privacy Policy</a><br>
      In accordance with that policy, we use Matomo to collect anonymised data on visits to, downloads from, and searches of this site.
      Contact <a href="mailto:zeller@embl.de">Georg Zeller</a> for details.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
