<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Example dataset with Confoundering • SIAMCAT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Example dataset with Confoundering">
<meta property="og:description" content="SIAMCAT">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><script type="text/javascript">
  var _paq = _paq || [];
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//tr-denbi.embl.de/heimdall/";
    _paq.push(['setTrackerUrl', u+'p.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'p.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SIAMCAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SIAMCAT_vignette.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/SIAMCAT_vignette.html">Introduction to SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_confounder.html">SIAMCAT example with confounders</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_holdout.html">Holdout testing with SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_meta.html">SIAMCAT machine learning meta-analysis</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_ml_pitfalls.html">Machine learning pitfalls</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_read-in.html">SIAMCAT input formats</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zellerlab/siamcat" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Example dataset with Confoundering</h1>
                        <h4 data-toc-skip class="author">Jakob Wirbel
and Georg Zeller</h4>
            <address class="author_afil">
      EMBL
Heidelberg<br><a class="author_email" href="mailto:#"></a><a href="mailto:georg.zeller@embl.de" class="email">georg.zeller@embl.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">Date last modified:
2020-11-11</h4>
      
      
      <div class="hidden name"><code>SIAMCAT_confounder.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="about-this-vignette">About This Vignette<a class="anchor" aria-label="anchor" href="#about-this-vignette"></a>
</h2>
<p>Here, we demonstrate the standard workflow of the
<code>SIAMCAT</code> package using as an example the dataset from <a href="https://www.ncbi.nlm.nih.gov/pubmed/24997787" class="external-link">Nielsen et
al. <em>Nat Biotechnol</em> 2014</a>. This dataset contains samples from
patients with inflammatory bowel disease and from controls.<br>
More importantly, these samples have been collected in two different
countries, Spain and Denmark. Together with technical differences
between these samples, this introduces a potent confounding factor into
the data. Here we are going to explore how <code>SIAMCAT</code> would
identify the confounding factor and what the results would be if you
account for the confounder or not.</p>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<p>First, we load the packages needed to perform the analyses.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://tidyverse.tidyverse.org" class="external-link">"tidyverse"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SIAMCAT"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">"ggpubr"</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="preparations">Preparations<a class="anchor" aria-label="anchor" href="#preparations"></a>
</h2>
<p>There are two different ways to access the data for our example
dataset. On the one hand, it is available through the
<code>curatedMetagenomicData</code> R package. However, using it here
would create many more dependencies for the <code>SIAMCAT</code>
package.<br>
Therefore, we here use data available through the EMBL cluster.</p>
<p>In the <code>SIAMCAT</code> paper, we performed the presented
analyses on the datasets available through
<code>curatedMetagenomicData</code>. If you want to reproduce the
analysis from the <code>SIAMCAT</code> paper, you can execute the code
chunks in the <code>curatedMetageomicData</code> section, otherwise
execute the code in the mOTUs2 section.</p>
<div class="section level3">
<h3 id="curatedmetagenomicsdata">curatedMetagenomicsData<a class="anchor" aria-label="anchor" href="#curatedmetagenomicsdata"></a>
</h3>
<p>First, we load the package:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/waldronlab/curatedMetagenomicData" class="external-link">"curatedMetagenomicData"</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="metadata">Metadata<a class="anchor" aria-label="anchor" href="#metadata"></a>
</h4>
<p>The data are part of the <code>combined_metadata</code></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.nielsen.full</span> <span class="op">&lt;-</span> <span class="va">combined_metadata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dataset_name</span><span class="op">==</span><span class="st">'NielsenHB_2014'</span><span class="op">)</span></span></code></pre></div>
<p>One thing we have to keep in mind are repeated samples per subject
(see also the <strong>Machine learning pitfalls</strong> vignette).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">meta.nielsen.full</span><span class="op">$</span><span class="va">subjectID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">meta.nielsen.full</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Some subjects (but not all) had been sampled multiple times.
Therefore, we want to remove repeated samplings for the same subject,
since the samples would otherwise not be indepdenent from another.</p>
<p>The visit number is encoded in the <code>sampleID</code>. Therefore,
we can use this information to extract when the samples have been taken
and use only the first visit for each subject.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen.full</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sampleID</span>, <span class="va">subjectID</span>, <span class="va">study_condition</span>, <span class="va">disease_subtype</span>,</span>
<span>        <span class="va">disease</span>, <span class="va">age</span>, <span class="va">country</span>, <span class="va">number_reads</span>, <span class="va">median_read_length</span>, <span class="va">BMI</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>visit<span class="op">=</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_extract.html" class="external-link">str_extract</a></span><span class="op">(</span><span class="va">sampleID</span>, <span class="st">'_[0-9]+$'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>visit<span class="op">=</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html" class="external-link">str_remove</a></span><span class="op">(</span><span class="va">visit</span>, <span class="st">'_'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>visit<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">visit</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>visit<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">visit</span><span class="op">)</span><span class="op">~</span><span class="fl">0</span>, <span class="cn">TRUE</span><span class="op">~</span><span class="va">visit</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">subjectID</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">visit</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">visit</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Sample_ID<span class="op">=</span><span class="va">sampleID</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Group<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">disease</span><span class="op">==</span><span class="st">'healthy'</span><span class="op">~</span><span class="st">'CTR'</span>,</span>
<span>                            <span class="cn">TRUE</span><span class="op">~</span><span class="va">disease_subtype</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now, we can restrict our metadata to samples with <code>UC</code> and
healthy control samples:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Group</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'UC'</span>, <span class="st">'CTR'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>As a last step, we can adjust the column names for the metadata so
that they agree with the data available from the EMBL cluster. Also, we
add rownames to the dataframe since <code>SIAMCAT</code> needs rownames
to match samples across metadata and features.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Country<span class="op">=</span><span class="va">country</span><span class="op">)</span></span>
<span><span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">meta.nielsen</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta.nielsen</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span><span class="op">$</span><span class="va">sampleID</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="taxonomic-profiles">Taxonomic Profiles<a class="anchor" aria-label="anchor" href="#taxonomic-profiles"></a>
</h4>
<p>We can load the taxonomic profiles generated with MetaPhlAn2 via the
<em>curatedMetagenomicsData</em> R package.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="st">'NielsenHB_2014.metaphlan_bugs_list.stool'</span></span>
<span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/curatedMetagenomicData.html" class="external-link">curatedMetagenomicData</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">x</span>, dryrun<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="va">feat</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">assayData</span><span class="op">$</span><span class="va">exprs</span></span></code></pre></div>
<p>The MetaPhlAn2 profiles contain information on different taxonomic
levels. Therefore, we want to restrict them to species-level profiles.
In a second step, we convert them into relative abundances (summing up
to 1) instead of using the percentages (summing up to 100) that
MetaPhlAn2 outputs.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="va">feat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span>, pattern<span class="op">=</span><span class="st">'s__'</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="va">feat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span>,pattern<span class="op">=</span><span class="st">'t__'</span>, invert <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span><span class="op">)</span></span></code></pre></div>
<p>The feature names are very long and may be a bit un-wieldy for
plotting later on, so we shorten them to only the species name:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_extract.html" class="external-link">str_extract</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span>, <span class="st">'s__.*$'</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="motus2-profiles">mOTUs2 Profiles<a class="anchor" aria-label="anchor" href="#motus2-profiles"></a>
</h3>
<p>Both metadata and features are available through the EMBL
cluster:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># base url for data download</span></span>
<span><span class="va">data.loc</span> <span class="op">&lt;-</span> <span class="st">'https://zenodo.org/api/files/d81e429c-870f-44e0-a44a-2a4aa541b6c1/'</span></span>
<span><span class="co">## metadata</span></span>
<span><span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data.loc</span>, <span class="st">'meta_Nielsen.tsv'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">396</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">9</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Delimiter:</span> "\t"</span></span>
<span><span class="co">## <span style="color: #BB0000;">chr</span> (5): Sample_ID, Individual_ID, Country, Gender, Group</span></span>
<span><span class="co">## <span style="color: #00BB00;">dbl</span> (4): Sampling_day, Age, BMI, Library_Size</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co"># also here, we have to remove repeated samplings and CD samples</span></span>
<span><span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Group</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'CTR'</span>, <span class="st">'UC'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Individual_ID</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Sampling_day</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">Sampling_day</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta.nielsen</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span><span class="op">$</span><span class="va">Sample_ID</span></span>
<span></span>
<span><span class="co">## features</span></span>
<span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data.loc</span>, <span class="st">'metaHIT_motus.tsv'</span><span class="op">)</span>, </span>
<span>                    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>, sep<span class="op">=</span><span class="st">'\t'</span>,</span>
<span>                    check.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="st">''</span>, comment.char <span class="op">=</span> <span class="st">''</span><span class="op">)</span></span>
<span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="va">feat</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span></span>
<span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/proportions.html" class="external-link">prop.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="siamcat-workflow-without-confounders">SIAMCAT Workflow (without Confounders)<a class="anchor" aria-label="anchor" href="#siamcat-workflow-without-confounders"></a>
</h2>
<div class="section level3">
<h3 id="the-siamcat-object">The SIAMCAT Object<a class="anchor" aria-label="anchor" href="#the-siamcat-object"></a>
</h3>
<p>Now, we have everything ready to create a <code>SIAMCAT</code> object
which stores the feature matrix, the meta-variables, and the label.
Here, the label is created using the information in the metadata.<br>
To demonstrate the normal <code>SIAMCAT</code> workflow, we remove the
confounding factor by only looking at samples from Spain. Below, we have
a look what would have happened if we had not removed them.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># remove Danish samples</span></span>
<span><span class="va">meta.nielsen.esp</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span><span class="op">[</span><span class="va">meta.nielsen</span><span class="op">$</span><span class="va">Country</span> <span class="op">==</span> <span class="st">'ESP'</span>,<span class="op">]</span></span>
<span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat</span>, meta<span class="op">=</span><span class="va">meta.nielsen.esp</span>, label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'UC'</span><span class="op">)</span></span>
<span><span class="co">## + starting create.label</span></span>
<span><span class="co">## Label used as case:</span></span>
<span><span class="co">##    UC</span></span>
<span><span class="co">## Label used as control:</span></span>
<span><span class="co">##    CTR</span></span>
<span><span class="co">## + finished create.label.from.metadata in 0.012 s</span></span>
<span><span class="co">## + starting validate.data</span></span>
<span><span class="co">## +++ checking overlap between labels and features</span></span>
<span><span class="co">## + Keeping labels of 128 sample(s).</span></span>
<span><span class="co">## +++ checking sample number per class</span></span>
<span><span class="co">## +++ checking overlap between samples and metadata</span></span>
<span><span class="co">## + finished validate.data in 0.268 s</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="filtering">Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h3>
<p>Now, we can filter feature with low overall abundance and
prevalence.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj</span>, cutoff<span class="op">=</span><span class="fl">1e-04</span>,</span>
<span>                            filter.method <span class="op">=</span> <span class="st">'abundance'</span><span class="op">)</span></span>
<span><span class="co">## Features successfully filtered</span></span>
<span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj</span>, cutoff<span class="op">=</span><span class="fl">0.05</span>,</span>
<span>                            filter.method<span class="op">=</span><span class="st">'prevalence'</span>,</span>
<span>                            feature.type <span class="op">=</span> <span class="st">'filtered'</span><span class="op">)</span></span>
<span><span class="co">## Features successfully filtered</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="association-plot">Association Plot<a class="anchor" aria-label="anchor" href="#association-plot"></a>
</h3>
<p>The <code>check.assocation</code> function calculates the
significance of enrichment and metrics of association (such as
generalized fold change and single-feature AUROC).</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/check.associations.html">check.associations</a></span><span class="op">(</span><span class="va">sc.obj</span>, log.n0 <span class="op">=</span> <span class="fl">1e-06</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/association.plot.html">association.plot</a></span><span class="op">(</span><span class="va">sc.obj</span>, fn.plot <span class="op">=</span> <span class="st">'./association_plot_nielsen.pdf'</span>, </span>
<span>                panels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'fc'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="association_plot_nielsen.png"></p>
</div>
<div class="section level3">
<h3 id="confounder-analysis">Confounder Analysis<a class="anchor" aria-label="anchor" href="#confounder-analysis"></a>
</h3>
<p>We can also check the supplied meta-variables for potential
confounding.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/check.confounders.html">check.confounders</a></span><span class="op">(</span><span class="va">sc.obj</span>, fn.plot <span class="op">=</span> <span class="st">'./confounders_nielsen.pdf'</span><span class="op">)</span></span>
<span><span class="co">## ++ remove metadata variables, since all subjects have the same value</span></span>
<span><span class="co">##  Country</span></span>
<span><span class="co">## Finished checking metadata for confounders, results plotted to: ./confounders_nielsen.pdf</span></span></code></pre></div>
<p><img src="confounders_nielsen.png"></p>
<p>The function produces one plot for each meta-variable. Here, we show
only the example of the body mass index (BMI). The BMI distributions
look very similar for both controls and UC cases, so it is unlikely that
the BMI would confound the analyses.</p>
</div>
<div class="section level3">
<h3 id="machine-learning-workflow">Machine Learning Workflow<a class="anchor" aria-label="anchor" href="#machine-learning-workflow"></a>
</h3>
<p>The machine learning workflow can be easily implemented in
<code>SIAMCAT</code>. It contains the following steps:</p>
<ul>
<li>Feature normalization</li>
<li>Data splitting for cross-validation</li>
<li>Model training</li>
<li>Making model predictions (on left-out data)</li>
<li>Evaluating model predictions (using AUROC and AUPRC)</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">sc.obj</span>, norm.method <span class="op">=</span> <span class="st">'log.std'</span>,</span>
<span>                            norm.param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>log.n0<span class="op">=</span><span class="fl">1e-06</span>, sd.min.q<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## Features normalized successfully.</span></span>
<span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span><span class="va">sc.obj</span>, num.folds <span class="op">=</span> <span class="fl">5</span>, num.resample <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">## Features splitted for cross-validation successfully.</span></span>
<span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj</span>, method<span class="op">=</span><span class="st">'lasso'</span><span class="op">)</span></span>
<span><span class="co">## Trained lasso models successfully.</span></span>
<span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span>
<span><span class="co">## Made predictions successfully.</span></span>
<span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span>
<span><span class="co">## Evaluated predictions successfully.</span></span></code></pre></div>
<div class="section level4">
<h4 id="model-evaluation-plot">Model Evaluation Plot<a class="anchor" aria-label="anchor" href="#model-evaluation-plot"></a>
</h4>
<p>The model evaluation plot will produce one plot with the ROC curve
and another one with the precision-recall curve (not shown here).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/model.evaluation.plot.html">model.evaluation.plot</a></span><span class="op">(</span><span class="va">sc.obj</span>, fn.plot <span class="op">=</span> <span class="st">'./eval_plot_nielsen.pdf'</span><span class="op">)</span></span>
<span><span class="co">## Plotted evaluation of predictions successfully to: ./eval_plot_nielsen.pdf</span></span></code></pre></div>
<p><img src="eval_plot_nielsen.png"></p>
</div>
<div class="section level4">
<h4 id="model-interpretation-plot">Model Interpretation Plot<a class="anchor" aria-label="anchor" href="#model-interpretation-plot"></a>
</h4>
<p>The model interpretation plot can give you additional information
about the trained machine learning model. It will show you:</p>
<ul>
<li>the feature importance as barplot,</li>
<li>the feature robustness (in how many of the models in the repeated
cross-validation this feature has been selected into the model),</li>
<li>the normalized feature abundances across samples as heatmap,</li>
<li>the optional metadata as heatmap below, and</li>
<li>a boxplot showing the proportion of the model weight that is
explained by<br>
the selected features.</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/model.interpretation.plot.html">model.interpretation.plot</a></span><span class="op">(</span><span class="va">sc.obj</span>, consens.thres <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>                            fn.plot <span class="op">=</span> <span class="st">'./interpret_nielsen.pdf'</span><span class="op">)</span></span>
<span><span class="co">## Successfully plotted model interpretation plot to: ./interpret_nielsen.pdf</span></span></code></pre></div>
<p><img src="interpretation_nielsen.png"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="confounder-analysis-1">Confounder Analysis<a class="anchor" aria-label="anchor" href="#confounder-analysis-1"></a>
</h2>
<p>As already mentioned above, the Nielsen dataset contains samples from
both Spain and Denmark. How would <code>SIAMCAT</code> have alerted us
to this?</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">meta.nielsen</span><span class="op">$</span><span class="va">Group</span>, <span class="va">meta.nielsen</span><span class="op">$</span><span class="va">Country</span><span class="op">)</span></span>
<span><span class="co">##      </span></span>
<span><span class="co">##       DNK ESP</span></span>
<span><span class="co">##   CTR 177  59</span></span>
<span><span class="co">##   UC    0  69</span></span></code></pre></div>
<div class="section level3">
<h3 id="country-confounder">Country Confounder<a class="anchor" aria-label="anchor" href="#country-confounder"></a>
</h3>
<p>First, we create a <code>SIAMCAT</code> object again, this time
including the Danish controls:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat</span>, meta<span class="op">=</span><span class="va">meta.nielsen</span>,</span>
<span>                        label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'UC'</span><span class="op">)</span></span>
<span><span class="co">## + starting create.label</span></span>
<span><span class="co">## Label used as case:</span></span>
<span><span class="co">##    UC</span></span>
<span><span class="co">## Label used as control:</span></span>
<span><span class="co">##    CTR</span></span>
<span><span class="co">## + finished create.label.from.metadata in 0.001 s</span></span>
<span><span class="co">## + starting validate.data</span></span>
<span><span class="co">## +++ checking overlap between labels and features</span></span>
<span><span class="co">## + Keeping labels of 291 sample(s).</span></span>
<span><span class="co">## + Removed 14 samples from the label object...</span></span>
<span><span class="co">## +++ checking sample number per class</span></span>
<span><span class="co">## +++ checking overlap between samples and metadata</span></span>
<span><span class="co">## + finished validate.data in 0.363 s</span></span>
<span><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, cutoff<span class="op">=</span><span class="fl">1e-04</span>,</span>
<span>                                filter.method <span class="op">=</span> <span class="st">'abundance'</span><span class="op">)</span></span>
<span><span class="co">## Features successfully filtered</span></span>
<span><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, cutoff<span class="op">=</span><span class="fl">0.05</span>,</span>
<span>                                filter.method<span class="op">=</span><span class="st">'prevalence'</span>,</span>
<span>                                feature.type <span class="op">=</span> <span class="st">'filtered'</span><span class="op">)</span></span>
<span><span class="co">## Features successfully filtered</span></span></code></pre></div>
<p>The confounder plot would show us that the meta-variable “country”
might be problematic:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/check.confounders.html">check.confounders</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, fn.plot <span class="op">=</span> <span class="st">'./confounders_dnk.pdf'</span><span class="op">)</span></span></code></pre></div>
<p><img src="confounders_dnk.png"></p>
</div>
<div class="section level3">
<h3 id="association-testing">Association Testing<a class="anchor" aria-label="anchor" href="#association-testing"></a>
</h3>
<p>First, we can use <code>SIAMCAT</code> to test for associations
including the Danish samples.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/check.associations.html">check.associations</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, log.n0 <span class="op">=</span> <span class="fl">1e-06</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> </span></code></pre></div>
<p>Confounders can lead to biases in association testing. After using
<code>SIAMCAT</code> to test for associations in both datasets (one time
including the Danish samples, the other time restricted to samples from
Spain only), we can extract the association metrics from both
<code>SIAMCAT</code> objects and compare them in a scatter plot.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">assoc.sp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/associations-methods.html">associations</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span>
<span><span class="va">assoc.sp</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">assoc.sp</span><span class="op">)</span></span>
<span><span class="va">assoc.sp_dnk</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/associations-methods.html">associations</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span></span>
<span><span class="va">assoc.sp_dnk</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">assoc.sp_dnk</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></span><span class="op">(</span><span class="va">assoc.sp</span>, <span class="va">assoc.sp_dnk</span>, by<span class="op">=</span><span class="st">'species'</span><span class="op">)</span></span>
<span><span class="va">df.plot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>highlight<span class="op">=</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">species</span>, <span class="st">'formicigenerans'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">p.adj.x</span><span class="op">)</span>, y<span class="op">=</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">p.adj.y</span><span class="op">)</span>, col<span class="op">=</span><span class="va">highlight</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">'Spanish samples only\n-log10(q)'</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'Spanish and Danish samples only\n-log10(q)'</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_line</a></span><span class="op">(</span>colour<span class="op">=</span><span class="st">'lightgrey'</span><span class="op">)</span>,</span>
<span>            aspect.ratio <span class="op">=</span> <span class="fl">1.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'darkgrey'</span>, <span class="st">'#D41645'</span><span class="op">)</span>, guide<span class="op">=</span><span class="st">'none'</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html" class="external-link">annotate</a></span><span class="op">(</span><span class="st">'text'</span>, x<span class="op">=</span><span class="fl">0.7</span>, y<span class="op">=</span><span class="fl">8</span>, label<span class="op">=</span><span class="st">'Dorea formicigenerans'</span><span class="op">)</span></span></code></pre></div>
<p><img src="SIAMCAT_confounder_files/figure-html/conf_assoc_plot-1.png" width="700"></p>
<p>This result shows that several species are only signficant if the
Danish control samples are included, but not when considering only the
Spanish samples.</p>
<p>As an example, we highlighted the species <em>“Dorea
formicigenerans”</em> in the plot above. The test is not significant in
the Spanish cohort, but is highly significant when the Danish samples
are included.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># extract information out of the siamcat object</span></span>
<span><span class="va">feat.dnk</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get.filt_feat.matrix.html">get.filt_feat.matrix</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span></span>
<span><span class="va">label.dnk</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/label-methods.html">label</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span><span class="op">$</span><span class="va">label</span></span>
<span><span class="va">country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/meta-methods.html">meta</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span><span class="op">$</span><span class="va">Country</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">country</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="../reference/meta-methods.html">meta</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>dorea<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">feat.dnk</span><span class="op">[</span></span>
<span>    <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat.dnk</span><span class="op">)</span>,<span class="st">'formicigenerans'</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">label.dnk</span><span class="op">)</span><span class="op">]</span> <span class="op">+</span> <span class="fl">1e-05</span><span class="op">)</span>,</span>
<span>    label<span class="op">=</span><span class="va">label.dnk</span>, country<span class="op">=</span><span class="va">country</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>label<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">label</span><span class="op">==</span><span class="st">'-1'</span><span class="op">~</span><span class="st">'CTR'</span>, <span class="cn">TRUE</span><span class="op">~</span><span class="st">"UC"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>x_value<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">country</span>, <span class="st">'_'</span>, <span class="va">label</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df.plot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">x_value</span>, y<span class="op">=</span><span class="va">dorea</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>outlier.shape <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.08</span>, stroke<span class="op">=</span><span class="fl">0</span>, alpha<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"log10(Dorea formicigenerans)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/stat_compare_means.html" class="external-link">stat_compare_means</a></span><span class="op">(</span>comparisons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'DNK_CTR'</span>, <span class="st">'ESP_CTR'</span><span class="op">)</span>,</span>
<span>                                                <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'DNK_CTR'</span>, <span class="st">'ESP_UC'</span><span class="op">)</span>,</span>
<span>                                                <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ESP_CTR'</span>, <span class="st">'ESP_UC'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="SIAMCAT_confounder_files/figure-html/dorea_plot-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="machine-learning">Machine Learning<a class="anchor" aria-label="anchor" href="#machine-learning"></a>
</h3>
<p>The results from the machine learning workflows can also be biased by
the differences between countries, leading to exaggerated performance
estimates.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, norm.method <span class="op">=</span> <span class="st">'log.std'</span>,</span>
<span>                                norm.param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>log.n0<span class="op">=</span><span class="fl">1e-06</span>, sd.min.q<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## Features normalized successfully.</span></span>
<span><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, num.folds <span class="op">=</span> <span class="fl">5</span>, num.resample <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">## Features splitted for cross-validation successfully.</span></span>
<span><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, method<span class="op">=</span><span class="st">'lasso'</span><span class="op">)</span></span>
<span><span class="co">## Trained lasso models successfully.</span></span>
<span><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span></span>
<span><span class="co">## Made predictions successfully.</span></span>
<span><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span></span>
<span><span class="co">## Evaluated predictions successfully.</span></span></code></pre></div>
<p>When we compare the performance of the two different models, the
model with the Danish and Spanish samples included seems to perform
better (higher AUROC value). However, the previous analysis suggests
that this performance estimate is biased and exaggerated because
differences between Spanish and Danish samples can be very large.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/model.evaluation.plot.html">model.evaluation.plot</a></span><span class="op">(</span><span class="st">"Spanish samples only"</span><span class="op">=</span><span class="va">sc.obj</span>,</span>
<span>                    <span class="st">"Danish and Spanish samples"</span><span class="op">=</span><span class="va">sc.obj.full</span>,</span>
<span>                    fn.plot <span class="op">=</span> <span class="st">'./eval_plot_dnk.pdf'</span><span class="op">)</span></span>
<span><span class="co">## Plotted evaluation of predictions successfully to: ./eval_plot_dnk.pdf</span></span></code></pre></div>
<p><img src="eval_plot_dnk.png"></p>
<p>To demonstrate how machine learning models can exploit this
confounding factor, we can train a model to distinguish between Spanish
and Danish control samples. As you can see, the model can distinguish
between the two countries with almost perfect accuracy.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.nielsen.country</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span><span class="op">[</span><span class="va">meta.nielsen</span><span class="op">$</span><span class="va">Group</span><span class="op">==</span><span class="st">'CTR'</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat</span>, meta<span class="op">=</span><span class="va">meta.nielsen.country</span>,</span>
<span>                            label<span class="op">=</span><span class="st">'Country'</span>, case<span class="op">=</span><span class="st">'ESP'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj.country</span>, cutoff<span class="op">=</span><span class="fl">1e-04</span>,</span>
<span>                            filter.method <span class="op">=</span> <span class="st">'abundance'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj.country</span>, cutoff<span class="op">=</span><span class="fl">0.05</span>,</span>
<span>                            filter.method<span class="op">=</span><span class="st">'prevalence'</span>,</span>
<span>                            feature.type <span class="op">=</span> <span class="st">'filtered'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">sc.obj.country</span>, norm.method <span class="op">=</span> <span class="st">'log.std'</span>,</span>
<span>                                    norm.param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>log.n0<span class="op">=</span><span class="fl">1e-06</span>,</span>
<span>                                        sd.min.q<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span><span class="va">sc.obj.country</span>, </span>
<span>                                    num.folds <span class="op">=</span> <span class="fl">5</span>, num.resample <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj.country</span>, method<span class="op">=</span><span class="st">'lasso'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.country</span><span class="op">)</span></span>
<span><span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.country</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="../reference/eval_data-methods.html">eval_data</a></span><span class="op">(</span><span class="va">sc.obj.country</span><span class="op">)</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span></span>
<span><span class="co">## Area under the curve: 0.9701</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ggpubr_0.6.0     SIAMCAT_2.3.3    phyloseq_1.42.0  mlr3_0.14.1     </span></span>
<span><span class="co">##  [5] lubridate_1.9.2  forcats_1.0.0    stringr_1.5.0    dplyr_1.1.0     </span></span>
<span><span class="co">##  [9] purrr_1.0.1      readr_2.1.4      tidyr_1.3.0      tibble_3.2.0    </span></span>
<span><span class="co">## [13] ggplot2_3.4.1    tidyverse_2.0.0  BiocStyle_2.26.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] uuid_1.1-0             backports_1.4.1        corrplot_0.92         </span></span>
<span><span class="co">##   [4] systemfonts_1.0.4      plyr_1.8.8             igraph_1.4.1          </span></span>
<span><span class="co">##   [7] splines_4.2.2          listenv_0.9.0          GenomeInfoDb_1.34.9   </span></span>
<span><span class="co">##  [10] gridBase_0.4-7         digest_0.6.31          foreach_1.5.2         </span></span>
<span><span class="co">##  [13] htmltools_0.5.4        lmerTest_3.1-3         fansi_1.0.4           </span></span>
<span><span class="co">##  [16] magrittr_2.0.3         checkmate_2.1.0        memoise_2.0.1         </span></span>
<span><span class="co">##  [19] cluster_2.1.4          tzdb_0.3.0             globals_0.16.2        </span></span>
<span><span class="co">##  [22] Biostrings_2.66.0      mlr3tuning_0.18.0      matrixStats_0.63.0    </span></span>
<span><span class="co">##  [25] vroom_1.6.1            timechange_0.2.0       pkgdown_2.0.7         </span></span>
<span><span class="co">##  [28] prettyunits_1.1.1      colorspace_2.1-0       textshaping_0.3.6     </span></span>
<span><span class="co">##  [31] xfun_0.37              crayon_1.5.2           RCurl_1.98-1.10       </span></span>
<span><span class="co">##  [34] jsonlite_1.8.4         lme4_1.1-32            survival_3.5-5        </span></span>
<span><span class="co">##  [37] iterators_1.0.14       ape_5.7-1              glue_1.6.2            </span></span>
<span><span class="co">##  [40] gtable_0.3.1           zlibbioc_1.44.0        XVector_0.38.0        </span></span>
<span><span class="co">##  [43] car_3.1-1              Rhdf5lib_1.20.0        shape_1.4.6           </span></span>
<span><span class="co">##  [46] BiocGenerics_0.44.0    abind_1.4-5            scales_1.2.1          </span></span>
<span><span class="co">##  [49] infotheo_1.2.0.1       DBI_1.1.3              rstatix_0.7.2         </span></span>
<span><span class="co">##  [52] Rcpp_1.0.10            progress_1.2.2         palmerpenguins_0.1.1  </span></span>
<span><span class="co">##  [55] bit_4.0.5              stats4_4.2.2           glmnet_4.1-6          </span></span>
<span><span class="co">##  [58] RColorBrewer_1.1-3     ellipsis_0.3.2         farver_2.1.1          </span></span>
<span><span class="co">##  [61] pkgconfig_2.0.3        sass_0.4.5             utf8_1.2.3            </span></span>
<span><span class="co">##  [64] labeling_0.4.2         tidyselect_1.2.0       rlang_1.1.0           </span></span>
<span><span class="co">##  [67] reshape2_1.4.4         PRROC_1.3.1            munsell_0.5.0         </span></span>
<span><span class="co">##  [70] tools_4.2.2            cachem_1.0.7           cli_3.6.0             </span></span>
<span><span class="co">##  [73] generics_0.1.3         ade4_1.7-22            broom_1.0.4           </span></span>
<span><span class="co">##  [76] evaluate_0.20          biomformat_1.26.0      fastmap_1.1.1         </span></span>
<span><span class="co">##  [79] yaml_2.3.7             ragg_1.2.5             bit64_4.0.5           </span></span>
<span><span class="co">##  [82] knitr_1.42             fs_1.6.1               lgr_0.4.4             </span></span>
<span><span class="co">##  [85] beanplot_1.3.1         bbotk_0.7.2            future_1.32.0         </span></span>
<span><span class="co">##  [88] nlme_3.1-162           paradox_0.11.0         compiler_4.2.2        </span></span>
<span><span class="co">##  [91] rstudioapi_0.14        curl_5.0.0             ggsignif_0.6.4        </span></span>
<span><span class="co">##  [94] bslib_0.4.2            stringi_1.7.12         highr_0.10            </span></span>
<span><span class="co">##  [97] desc_1.4.2             lattice_0.20-45        Matrix_1.5-3          </span></span>
<span><span class="co">## [100] nloptr_2.0.3           vegan_2.6-4            permute_0.9-7         </span></span>
<span><span class="co">## [103] multtest_2.54.0        vctrs_0.5.2            pillar_1.8.1          </span></span>
<span><span class="co">## [106] lifecycle_1.0.3        rhdf5filters_1.10.0    BiocManager_1.30.20   </span></span>
<span><span class="co">## [109] jquerylib_0.1.4        LiblineaR_2.10-22      data.table_1.14.8     </span></span>
<span><span class="co">## [112] bitops_1.0-7           R6_2.5.1               bookdown_0.33         </span></span>
<span><span class="co">## [115] gridExtra_2.3          mlr3misc_0.11.0        IRanges_2.32.0        </span></span>
<span><span class="co">## [118] parallelly_1.34.0      codetools_0.2-19       boot_1.3-28.1         </span></span>
<span><span class="co">## [121] MASS_7.3-58.3          rhdf5_2.42.0           mlr3learners_0.5.6    </span></span>
<span><span class="co">## [124] rprojroot_2.0.3        withr_2.5.0            S4Vectors_0.36.2      </span></span>
<span><span class="co">## [127] GenomeInfoDbData_1.2.9 mgcv_1.8-42            parallel_4.2.2        </span></span>
<span><span class="co">## [130] hms_1.1.2              grid_4.2.2             minqa_1.2.5           </span></span>
<span><span class="co">## [133] rmarkdown_2.20         carData_3.0-5          pROC_1.18.0           </span></span>
<span><span class="co">## [136] numDeriv_2016.8-1.1    Biobase_2.58.0</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Konrad Zych, Jakob Wirbel, Georg Zeller.</p>
</div>

<div class="privacy">
  <p><strong>Privacy:</strong> Usage of this site follows
      <a href="https://www.embl.de/aboutus/privacy_policy/index.html" class="external-link">EMBL’s Privacy Policy</a><br>
      In accordance with that policy, we use Matomo to collect anonymised data on visits to, downloads from, and searches of this site.
      Contact <a href="mailto:zeller@embl.de">Georg Zeller</a> for details.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
