<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Machine learning pitfalls â€¢ SIAMCAT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Machine learning pitfalls">
<meta property="og:description" content="SIAMCAT">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><script type="text/javascript">
  var _paq = _paq || [];
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//tr-denbi.embl.de/heimdall/";
    _paq.push(['setTrackerUrl', u+'p.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'p.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SIAMCAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SIAMCAT_vignette.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/SIAMCAT_vignette.html">Introduction to SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_confounder.html">SIAMCAT example with confounders</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_holdout.html">Holdout testing with SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_meta.html">SIAMCAT machine learning meta-analysis</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_ml_pitfalls.html">Machine learning pitfalls</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_read-in.html">SIAMCAT input formats</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zellerlab/siamcat" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Machine learning pitfalls</h1>
                        <h4 data-toc-skip class="author">Jakob Wirbel
and Georg Zeller</h4>
            <address class="author_afil">
      EMBL
Heidelberg<br><a class="author_email" href="mailto:#"></a><a href="mailto:georg.zeller@embl.de" class="email">georg.zeller@embl.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">Date last modified:
2020-11-07</h4>
      
      
      <div class="hidden name"><code>SIAMCAT_ml_pitfalls.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="about-this-vignette">About This Vignette<a class="anchor" aria-label="anchor" href="#about-this-vignette"></a>
</h2>
<p>In this vignette, we want to explore two pitfalls for machine
learning analysis that can lead to overly optimistic performance
estimates.<br>
When setting up cross-validation workflows, the main objective is
usually to estimate how well a trained model would perform on external
data, which is specifically important when considering biomarker
discovery. However, more complex workflows involving feature selection
or time-course data can be challenging to setup correctly. Incorrect
workflows in which information leaks from the test to the training data
can lead to overfitting and poor generalization to external
datasets.<br>
Here, we focus on supervised feature selection and the naive splitting
of dependent data.</p>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<p>First, we load the packages needed to perform the analyses.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://tidyverse.tidyverse.org" class="external-link">"tidyverse"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SIAMCAT"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="supervised-feature-selection">Supervised Feature Selection<a class="anchor" aria-label="anchor" href="#supervised-feature-selection"></a>
</h2>
<p>Supervised feature selection means that the label information is
taken into account before the cross-validation split. Within this
procedure, the features are selected if they are associated with the
label (for example after differential abundance testing), using the
complete dataset for the calculation of feature association and leaving
no data aside for unbiased model evaluation.<br>
A correct way to perform feature selection would be to nest the
selection step into the cross-validation procedure. That means that the
calculation of feature association is performed for each training fold
separately.</p>
<div class="section level3">
<h3 id="load-the-data">Load the Data<a class="anchor" aria-label="anchor" href="#load-the-data"></a>
</h3>
<p>As an example, we are going to use two datasets of colorectal cancer
(CRC) which are available through the
<code>curatedMetagenomicData</code> package.<br>
Since the model trainig procedure takes a long time, this vignette is
not evaluated upon build of the package, but if you execute the code
chunks for yourself, you should get similar results.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/waldronlab/curatedMetagenomicData" class="external-link">"curatedMetagenomicData"</a></span><span class="op">)</span></span></code></pre></div>
<p>First, we are going to load the dataset from <a href="https://doi.org/10.1038/s41591-019-0405-7" class="external-link">Thomas et al</a> as
training dataset.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="st">'ThomasAM_2018a.metaphlan_bugs_list.stool'</span></span>
<span><span class="va">feat.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/curatedMetagenomicData.html" class="external-link">curatedMetagenomicData</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">x</span>, dryrun<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">feat.t</span> <span class="op">&lt;-</span> <span class="va">feat.t</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">assayData</span><span class="op">$</span><span class="va">exprs</span></span>
<span><span class="co"># clean up metaphlan profiles to contain only species-level abundances</span></span>
<span><span class="va">feat.t</span> <span class="op">&lt;-</span> <span class="va">feat.t</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat.t</span><span class="op">)</span>, pattern<span class="op">=</span><span class="st">'s__'</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">feat.t</span> <span class="op">&lt;-</span> <span class="va">feat.t</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat.t</span><span class="op">)</span>,pattern<span class="op">=</span><span class="st">'t__'</span>, invert <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">feat.t</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">feat.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">feat.t</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span><span class="op">)</span></span></code></pre></div>
<p>As an external dataset, we are going to use the data from <a href="https://doi.org/10.15252/msb.20145645" class="external-link">Zeller et al.</a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="st">'ZellerG_2014.metaphlan_bugs_list.stool'</span></span>
<span><span class="va">feat.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/curatedMetagenomicData.html" class="external-link">curatedMetagenomicData</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">x</span>, dryrun<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">feat.z</span> <span class="op">&lt;-</span> <span class="va">feat.z</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">assayData</span><span class="op">$</span><span class="va">exprs</span></span>
<span><span class="co"># clean up metaphlan profiles to contain only species-level abundances</span></span>
<span><span class="va">feat.z</span> <span class="op">&lt;-</span> <span class="va">feat.z</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat.z</span><span class="op">)</span>, pattern<span class="op">=</span><span class="st">'s__'</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">feat.z</span> <span class="op">&lt;-</span> <span class="va">feat.z</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat.z</span><span class="op">)</span>,pattern<span class="op">=</span><span class="st">'t__'</span>, invert <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">feat.z</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">feat.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">feat.z</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span><span class="op">)</span></span></code></pre></div>
<p>We can also extract the corresponding metadata from the
<code>combined_metadata</code> object which is part of the
<code>curatedMetagenomicData</code> package.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.t</span> <span class="op">&lt;-</span> <span class="va">combined_metadata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dataset_name</span> <span class="op">==</span> <span class="st">'ThomasAM_2018a'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">study_condition</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'control'</span>, <span class="st">'CRC'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta.t</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">meta.t</span><span class="op">$</span><span class="va">sampleID</span></span>
<span><span class="va">meta.z</span> <span class="op">&lt;-</span> <span class="va">combined_metadata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">dataset_name</span> <span class="op">==</span> <span class="st">'ZellerG_2014'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">study_condition</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'control'</span>, <span class="st">'CRC'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta.z</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">meta.z</span><span class="op">$</span><span class="va">sampleID</span></span></code></pre></div>
<p>The MetaPhlAn2 profiler used for the profiles outputs only species
which are present in the dataset. Therefore, we can have the case that
there are species in the matrix for <code>ThomasAM_2018</code> which are
not present in the matrix for <code>ZellerG_2014</code> and vice verse.
In order to use them as training and external test set for
<code>SIAMCAT</code>, we have to first make sure that the set of
features for both datasets overlap completely (see also the
<strong>Holdout Testing with SIAMCAT</strong> vignette).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">species.union</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">union</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat.t</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat.z</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># add Zeller_2014-only species to the Thomas_2018 matrix</span></span>
<span><span class="va">add.species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">species.union</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat.t</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">feat.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">feat.t</span>, </span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">0</span>, nrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">add.species</span><span class="op">)</span>, ncol<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">feat.t</span><span class="op">)</span>,</span>
<span>                dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">add.species</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">feat.t</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add Thomas_2018-only species to the Zeller_2014 matrix</span></span>
<span><span class="va">add.species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">species.union</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">feat.z</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">feat.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">feat.z</span>, </span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">0</span>, nrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">add.species</span><span class="op">)</span>, ncol<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">feat.z</span><span class="op">)</span>,</span>
<span>                dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">add.species</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">feat.z</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now, we are ready to start the model training process. For this, we
chose three different feature selection cutoffs and prepare a tibble to
hold the results:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fs.cutoff</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">20</span>, <span class="fl">100</span>, <span class="fl">250</span><span class="op">)</span></span>
<span></span>
<span><span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>cutoff<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span>, type<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span>, </span>
<span>                    study.test<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span>, AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/double.html" class="external-link">double</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="train-model-without-feature-selection">Train Model without Feature Selection<a class="anchor" aria-label="anchor" href="#train-model-without-feature-selection"></a>
</h3>
<p>First, we will train a model without any feature selection, using all
the available features. We add it to the results matrix twice (both with
<code>correct</code> and <code>incorrect</code>) for easier plotting
later.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.t</span>, meta<span class="op">=</span><span class="va">meta.t</span>,</span>
<span>                    label<span class="op">=</span><span class="st">'study_condition'</span>, case<span class="op">=</span><span class="st">'CRC'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj.t</span>, filter.method <span class="op">=</span> <span class="st">'prevalence'</span>,</span>
<span>                            cutoff <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="va">sc.obj.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">sc.obj.t</span>, norm.method <span class="op">=</span> <span class="st">'log.std'</span>,</span>
<span>                                norm.param<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>log.n0<span class="op">=</span><span class="fl">1e-05</span>, sd.min.q<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sc.obj.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span><span class="va">sc.obj.t</span>,</span>
<span>                                num.folds <span class="op">=</span> <span class="fl">10</span>, num.resample <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">sc.obj.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj.t</span>, method<span class="op">=</span><span class="st">'lasso'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.t</span><span class="op">)</span></span>
<span><span class="va">sc.obj.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.t</span><span class="op">)</span></span>
<span></span>
<span><span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>cutoff<span class="op">=</span><span class="st">'full'</span>, type<span class="op">=</span><span class="st">'correct'</span>, </span>
<span>            study.test<span class="op">=</span><span class="st">'Thomas_2018'</span>, </span>
<span>            AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sc.obj.t</span><span class="op">@</span><span class="va">eval_data</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>cutoff<span class="op">=</span><span class="st">'full'</span>, type<span class="op">=</span><span class="st">'incorrect'</span>, study.test<span class="op">=</span><span class="st">'Thomas_2018'</span>, </span>
<span>            AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sc.obj.t</span><span class="op">@</span><span class="va">eval_data</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<p>We then also apply the model to the external dataset and record the
generalization to another dataset:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.z</span>, meta<span class="op">=</span><span class="va">meta.z</span>,</span>
<span>                    label<span class="op">=</span><span class="st">'study_condition'</span>, case<span class="op">=</span><span class="st">'CRC'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.t</span>, <span class="va">sc.obj.z</span><span class="op">)</span></span>
<span><span class="va">sc.obj.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.z</span><span class="op">)</span></span>
<span><span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>cutoff<span class="op">=</span><span class="st">'full'</span>, type<span class="op">=</span><span class="st">'correct'</span>, </span>
<span>            study.test<span class="op">=</span><span class="st">'Zeller_2014'</span>, </span>
<span>            AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sc.obj.z</span><span class="op">@</span><span class="va">eval_data</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>cutoff<span class="op">=</span><span class="st">'full'</span>, type<span class="op">=</span><span class="st">'incorrect'</span>, </span>
<span>            study.test<span class="op">=</span><span class="st">'Zeller_2014'</span>, </span>
<span>            AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sc.obj.z</span><span class="op">@</span><span class="va">eval_data</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level3">
<h3 id="incorrect-procedure-train-with-supervised-feature-selection">Incorrect Procedure: Train with Supervised Feature Selection<a class="anchor" aria-label="anchor" href="#incorrect-procedure-train-with-supervised-feature-selection"></a>
</h3>
<p>For the incorrect feature selection procedure, we can test the
features for differential abundance, using the complete dataset, and
then chose the top associated features.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/check.associations.html">check.associations</a></span><span class="op">(</span><span class="va">sc.obj.t</span>, detect.lim <span class="op">=</span> <span class="fl">1e-05</span>,</span>
<span>                                fn.plot <span class="op">=</span> <span class="st">'assoc_plot.pdf'</span><span class="op">)</span></span>
<span><span class="va">mat.assoc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/associations-methods.html">associations</a></span><span class="op">(</span><span class="va">sc.obj.t</span><span class="op">)</span></span>
<span><span class="va">mat.assoc</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">mat.assoc</span><span class="op">)</span></span>
<span><span class="co"># sort by p-value</span></span>
<span><span class="va">mat.assoc</span> <span class="op">&lt;-</span> <span class="va">mat.assoc</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">p.val</span><span class="op">)</span></span></code></pre></div>
<p>Based on the P values from the <code>check.association</code>
function, we now chose <code>X</code> number of features on which to
train the model.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">x</span> <span class="kw">in</span> <span class="va">fs.cutoff</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="co"># select x number of features based on p-value ranking</span></span>
<span>    <span class="va">feat.train.red</span> <span class="op">&lt;-</span> <span class="va">feat.t</span><span class="op">[</span><span class="va">mat.assoc</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                                <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                                <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">species</span><span class="op">)</span>,<span class="op">]</span></span>
<span>    <span class="va">sc.obj.t.fs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.train.red</span>, meta<span class="op">=</span><span class="va">meta.t</span>,</span>
<span>                            label<span class="op">=</span><span class="st">'study_condition'</span>, case<span class="op">=</span><span class="st">'CRC'</span><span class="op">)</span></span>
<span>    <span class="co"># normalize the features without filtering</span></span>
<span>    <span class="va">sc.obj.t.fs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span>, norm.method <span class="op">=</span> <span class="st">'log.std'</span>,</span>
<span>        norm.param<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>log.n0<span class="op">=</span><span class="fl">1e-05</span>,sd.min.q<span class="op">=</span><span class="fl">0</span><span class="op">)</span>,feature.type <span class="op">=</span> <span class="st">'original'</span><span class="op">)</span></span>
<span>    <span class="co"># take the same cross validation split as before</span></span>
<span>    <span class="fu"><a href="../reference/data_split-methods.html">data_split</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/data_split-methods.html">data_split</a></span><span class="op">(</span><span class="va">sc.obj.t</span><span class="op">)</span></span>
<span>    <span class="co"># train</span></span>
<span>    <span class="va">sc.obj.t.fs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span>, method <span class="op">=</span> <span class="st">'lasso'</span><span class="op">)</span></span>
<span>    <span class="co"># make predictions</span></span>
<span>    <span class="va">sc.obj.t.fs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span><span class="op">)</span></span>
<span>    <span class="co"># evaluate predictions and record the result</span></span>
<span>    <span class="va">sc.obj.t.fs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span><span class="op">)</span></span>
<span>    <span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>cutoff<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, type<span class="op">=</span><span class="st">'incorrect'</span>, </span>
<span>                study.test<span class="op">=</span><span class="st">'Thomas_2018'</span>,</span>
<span>                AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span><span class="op">@</span><span class="va">eval_data</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="co"># apply to the external dataset and record the result</span></span>
<span>    <span class="va">sc.obj.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.z</span>, meta<span class="op">=</span><span class="va">meta.z</span>,</span>
<span>                        label<span class="op">=</span><span class="st">'study_condition'</span>, case<span class="op">=</span><span class="st">'CRC'</span><span class="op">)</span></span>
<span>    <span class="va">sc.obj.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span>, <span class="va">sc.obj.z</span><span class="op">)</span></span>
<span>    <span class="va">sc.obj.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.z</span><span class="op">)</span></span>
<span>    <span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>cutoff<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, type<span class="op">=</span><span class="st">'incorrect'</span>, </span>
<span>                study.test<span class="op">=</span><span class="st">'Zeller_2014'</span>, </span>
<span>                AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sc.obj.z</span><span class="op">@</span><span class="va">eval_data</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="correct-procedure-train-with-nested-feature-selection">Correct Procedure: Train with Nested Feature Selection<a class="anchor" aria-label="anchor" href="#correct-procedure-train-with-nested-feature-selection"></a>
</h3>
<p>Feature selection can be performed correctly if it is nested within
the cross-validation procedure. We can do it using <code>SIAMCAT</code>
by specifying the <code>perform.fs</code> parameter in the
<code>train.model</code> function.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">x</span> <span class="kw">in</span> <span class="va">fs.cutoff</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="co"># train using the original SIAMCAT object </span></span>
<span>    <span class="co"># with correct version of feature selection</span></span>
<span>    <span class="va">sc.obj.t.fs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj.t</span>, method <span class="op">=</span> <span class="st">'lasso'</span>, perform.fs <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        param.fs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>thres.fs <span class="op">=</span> <span class="va">x</span>,method.fs <span class="op">=</span> <span class="st">"AUC"</span>,direction<span class="op">=</span><span class="st">'absolute'</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="co"># make predictions</span></span>
<span>    <span class="va">sc.obj.t.fs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span><span class="op">)</span></span>
<span>    <span class="co"># evaluate predictions and record the result</span></span>
<span>    <span class="va">sc.obj.t.fs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span><span class="op">)</span></span>
<span>    <span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>cutoff<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, type<span class="op">=</span><span class="st">'correct'</span>, </span>
<span>                study.test<span class="op">=</span><span class="st">'Thomas_2018'</span>,</span>
<span>                AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span><span class="op">@</span><span class="va">eval_data</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="co"># apply to the external dataset and record the result</span></span>
<span>    <span class="va">sc.obj.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.z</span>, meta<span class="op">=</span><span class="va">meta.z</span>,</span>
<span>                        label<span class="op">=</span><span class="st">'study_condition'</span>, case<span class="op">=</span><span class="st">'CRC'</span><span class="op">)</span></span>
<span>    <span class="va">sc.obj.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.t.fs</span>, <span class="va">sc.obj.z</span><span class="op">)</span></span>
<span>    <span class="va">sc.obj.z</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.z</span><span class="op">)</span></span>
<span>    <span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>cutoff<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, type<span class="op">=</span><span class="st">'correct'</span>, </span>
<span>                study.test<span class="op">=</span><span class="st">'Zeller_2014'</span>, </span>
<span>                AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">sc.obj.z</span><span class="op">@</span><span class="va">eval_data</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-the-results">Plot the Results<a class="anchor" aria-label="anchor" href="#plot-the-results"></a>
</h3>
<p>Now, we can plot the resulting performance estimates for the
cross-validation and the external validation as well:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># facetting for plotting</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>split<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">study.test</span><span class="op">==</span><span class="st">"Thomas_2018"</span><span class="op">~</span></span>
<span>                            <span class="st">'Cross validation (Thomas 2018)'</span>,</span>
<span>                        <span class="cn">TRUE</span><span class="op">~</span><span class="st">"External validation (Zeller 2014)"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># convert to factor to enforce ordering</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>cutoff<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">cutoff</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">fs.cutoff</span>, <span class="st">'full'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">cutoff</span>, y<span class="op">=</span><span class="va">AUC</span>, col<span class="op">=</span><span class="va">type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group<span class="op">=</span><span class="va">type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="op">~</span><span class="va">split</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">1</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">'Features selected'</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'AUROC'</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'correct'</span><span class="op">=</span><span class="st">'blue'</span>, <span class="st">'incorrect'</span><span class="op">=</span><span class="st">'red'</span><span class="op">)</span>,</span>
<span>            name<span class="op">=</span><span class="st">'Feature selection procedure'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, legend.position <span class="op">=</span> <span class="st">'bottom'</span><span class="op">)</span></span></code></pre></div>
<p><img src="SIAMCAT_ml_pitfalls_files/figure-html/plot_auroc-1.png" width="700"></p>
<p>As you can see, the incorrect feature selection procedure leads to
inflated AUROC values but lower generalization to a truly external
dataset, especially when very few features were selected. In contrast,
the correct procedure gives a lower cross-validation results but a
better estimation for how the model would perform on external data.</p>
</div>
</div>
<div class="section level2">
<h2 id="naive-splitting-of-dependent-data">Naive Splitting of Dependent Data<a class="anchor" aria-label="anchor" href="#naive-splitting-of-dependent-data"></a>
</h2>
<p>Another issue in machine learning workflows can occur when samples
are not independent. For example, microbiome samples taken from the same
individual at different time points are usually more similar to each
other than to samples from other individuals. If these samples are split
randomly in a naive cross-validation procedure, the case could arise
that samples from the same individual will end up in the training and
the test fold. In this case, the model would learn to generalize across
time-points for the same individual compared to the desired model that
should learn to distinguish the label across individuals.<br>
To avoid this issue, dependent measurements need to be blocked during
cross-validation, to ensure that samples within the same block will stay
in the same fold (for training and testing).</p>
<div class="section level3">
<h3 id="load-the-data-1">Load the Data<a class="anchor" aria-label="anchor" href="#load-the-data-1"></a>
</h3>
<p>As an example, we are going to use several datasets of Crohnâ€™s
disease (CD) which are available through the EMBL cluster. The data have
already been filtered and cleaned.<br>
Since the model training would take again quite a long time, this part
of the vignette is not evaluated upon building of the package, but you
should be able to execute it yourself.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data.loc</span> <span class="op">&lt;-</span> <span class="st">'https://zenodo.org/api/files/d81e429c-870f-44e0-a44a-2a4aa541b6c1/'</span></span>
<span></span>
<span><span class="co"># metadata</span></span>
<span><span class="va">meta.all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data.loc</span>, <span class="st">'meta_all_cd.tsv'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1597</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">6</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">â”€â”€</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Delimiter:</span> "\t"</span></span>
<span><span class="co">## <span style="color: #BB0000;">chr</span> (4): Sample_ID, Group, Individual_ID, Study</span></span>
<span><span class="co">## <span style="color: #00BB00;">dbl</span> (2): Library_Size, Timepoint</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">â„¹</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">â„¹</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span></span>
<span><span class="co"># features</span></span>
<span><span class="va">feat.motus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data.loc</span>, <span class="st">'feat_rel_filt_cd.tsv'</span><span class="op">)</span>,</span>
<span>                        sep<span class="op">=</span><span class="st">'\t'</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                        check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>When we look at the number of samples and number of individuals, we
see that that there are several samples per individual for example in
the <code>HMP2</code> study.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">meta.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Study</span>, <span class="va">Group</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>n.all<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">'drop'</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="va">meta.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Study</span>, <span class="va">Group</span>, <span class="va">Individual_ID</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Study</span>, <span class="va">Group</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>n.indi<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>,  .groups<span class="op">=</span><span class="st">'drop'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span></span>
<span><span class="co">## Joining with `by = join_by(Study, Group)`</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 10 Ã— 4</span></span></span>
<span><span class="co">##    Study         Group n.all n.indi</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> Franzosa_2019 CD       88     88</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Franzosa_2019 CTR      56     56</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> HMP2          CD      583     50</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> HMP2          CTR     357     26</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> He_2017       CD       49     49</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> He_2017       CTR      53     53</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> Lewis_2015    CD      294     85</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Lewis_2015    CTR      25     25</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> metaHIT       CD       21     13</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> metaHIT       CTR      71     59</span></span></code></pre></div>
<p>Therefore, we are going to train a model on the <code>HMP2</code>
study. However, the number of samples per individual varies quite a lot
across samples, therefore we want to randomly select a set of 5 samples
per individual:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Study</span><span class="op">==</span><span class="st">'HMP2'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Individual_ID</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>n<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">'drop'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="SIAMCAT_ml_pitfalls_files/figure-html/hmp_samples-1.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># sample 5 samples per individual</span></span>
<span><span class="va">meta.train</span> <span class="op">&lt;-</span> <span class="va">meta.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Study</span><span class="op">==</span><span class="st">'HMP2'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Individual_ID</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/sample_n.html" class="external-link">sample_n</a></span><span class="op">(</span><span class="fl">5</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta.train</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">meta.train</span><span class="op">$</span><span class="va">Sample_ID</span></span></code></pre></div>
<p>For evaluation, we only want a single sample per individual,
therefore we can create a new matrix removing repeated samples for the
other studies:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.ind</span> <span class="op">&lt;-</span> <span class="va">meta.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Individual_ID</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Timepoint</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">Timepoint</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Lastly, we can already create a tibble to hold the resulting AUROC
values:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>type<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span>, study.test<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span>, AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/double.html" class="external-link">double</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="train-with-naive-cross-validation">Train with Naive Cross-validation<a class="anchor" aria-label="anchor" href="#train-with-naive-cross-validation"></a>
</h3>
<p>The naive way to split samples for cross-validation does not take
into account the dependency between samples. Therefore, the pipeline
would look basically like this:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.motus</span>, meta<span class="op">=</span><span class="va">meta.train</span>,</span>
<span>                    label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'CD'</span><span class="op">)</span></span>
<span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">sc.obj</span>, norm.method <span class="op">=</span> <span class="st">'log.std'</span>,</span>
<span>    norm.param<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>log.n0<span class="op">=</span><span class="fl">1e-05</span>,sd.min.q<span class="op">=</span><span class="fl">1</span><span class="op">)</span>,feature.type <span class="op">=</span> <span class="st">'original'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.naive</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span><span class="va">sc.obj</span>, num.folds <span class="op">=</span> <span class="fl">10</span>, num.resample <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">sc.obj.naive</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj.naive</span>, method<span class="op">=</span><span class="st">'lasso'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.naive</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.naive</span><span class="op">)</span></span>
<span><span class="va">sc.obj.naive</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.naive</span><span class="op">)</span></span>
<span><span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">'naive'</span>, study.test<span class="op">=</span><span class="st">'HMP2'</span>, </span>
<span>        AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="../reference/eval_data-methods.html">eval_data</a></span><span class="op">(</span><span class="va">sc.obj.naive</span><span class="op">)</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="train-with-blocked-cross-validation">Train with Blocked Cross-validation<a class="anchor" aria-label="anchor" href="#train-with-blocked-cross-validation"></a>
</h3>
<p>The correct way to to take into account repeated samples would be to
block the cross-validation procedure by individuals. This way, samples
from the same individual will always end up in the same fold. This can
be performed in <code>SIAMCAT</code> by specifying the
<code>inseparable</code> parameter in the <code>create.data.split</code>
function:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj.block</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span><span class="va">sc.obj</span>, num.folds <span class="op">=</span> <span class="fl">10</span>, num.resample <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                                inseparable <span class="op">=</span> <span class="st">'Individual_ID'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.block</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj.block</span>, method<span class="op">=</span><span class="st">'lasso'</span><span class="op">)</span></span>
<span><span class="va">sc.obj.block</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.block</span><span class="op">)</span></span>
<span><span class="va">sc.obj.block</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.block</span><span class="op">)</span></span>
<span><span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">'blocked'</span>, study.test<span class="op">=</span><span class="st">'HMP2'</span>, </span>
<span>        AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="../reference/eval_data-methods.html">eval_data</a></span><span class="op">(</span><span class="va">sc.obj.block</span><span class="op">)</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="apply-to-external-datasets">Apply to External Datasets<a class="anchor" aria-label="anchor" href="#apply-to-external-datasets"></a>
</h3>
<p>Now, we can apply both models to external datasets and record the
resulting accuracy:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">meta.all</span><span class="op">$</span><span class="va">Study</span><span class="op">)</span>, <span class="st">'HMP2'</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">meta.test</span> <span class="op">&lt;-</span> <span class="va">meta.ind</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Study</span><span class="op">==</span><span class="va">i</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta.test</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">meta.test</span><span class="op">$</span><span class="va">Sample_ID</span></span>
<span>    <span class="co"># apply naive model</span></span>
<span>    <span class="va">sc.obj.test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.motus</span>, meta<span class="op">=</span><span class="va">meta.test</span>, </span>
<span>                            label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'CD'</span><span class="op">)</span></span>
<span>    <span class="va">sc.obj.test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.naive</span>, <span class="va">sc.obj.test</span><span class="op">)</span></span>
<span>    <span class="va">sc.obj.test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.test</span><span class="op">)</span></span>
<span>    <span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">'naive'</span>, study.test<span class="op">=</span><span class="va">i</span>,</span>
<span>            AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="../reference/eval_data-methods.html">eval_data</a></span><span class="op">(</span><span class="va">sc.obj.test</span><span class="op">)</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="co"># apply blocked model</span></span>
<span>    <span class="va">sc.obj.test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.motus</span>, meta<span class="op">=</span><span class="va">meta.test</span>, </span>
<span>                            label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'CD'</span><span class="op">)</span></span>
<span>    <span class="va">sc.obj.test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.block</span>, <span class="va">sc.obj.test</span><span class="op">)</span></span>
<span>    <span class="va">sc.obj.test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.test</span><span class="op">)</span></span>
<span>    <span class="va">auroc.all</span> <span class="op">&lt;-</span> <span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://tibble.tidyverse.org/reference/add_row.html" class="external-link">add_row</a></span><span class="op">(</span>type<span class="op">=</span><span class="st">'blocked'</span>, study.test<span class="op">=</span><span class="va">i</span>,</span>
<span>                AUC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="../reference/eval_data-methods.html">eval_data</a></span><span class="op">(</span><span class="va">sc.obj.test</span><span class="op">)</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-the-results-1">Plot the Results<a class="anchor" aria-label="anchor" href="#plot-the-results-1"></a>
</h3>
<p>Now, we can compare the resulting AUROC values between the two
different approaches:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">auroc.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># convert to factor to enforce ordering</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>type<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">type</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'naive'</span>, <span class="st">'blocked'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># facetting for plotting</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>CV<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">study.test</span><span class="op">==</span><span class="st">'HMP2'</span><span class="op">~</span><span class="st">'CV'</span>, </span>
<span>                        <span class="cn">TRUE</span><span class="op">~</span><span class="st">'External validation'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">study.test</span>, y<span class="op">=</span><span class="va">AUC</span>, fill<span class="op">=</span><span class="va">type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">'identity'</span>, position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span><span class="op">)</span>, col<span class="op">=</span><span class="st">'black'</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'red'</span>, <span class="st">'blue'</span><span class="op">)</span>, name<span class="op">=</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="op">~</span><span class="va">CV</span>, space <span class="op">=</span> <span class="st">'free'</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">'AUROC'</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.8</span>, <span class="fl">0.8</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="SIAMCAT_ml_pitfalls_files/figure-html/plot_results_db-1.png" width="700"></p>
<p>As you can see, the naive cross-validation procedure leads to a
inflated performance estimation compared to the blocked
cross-validation. However, when assessing generalization to truly
external datasets, the blocked procedure results in better
performance.</p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] SIAMCAT_2.3.3    phyloseq_1.42.0  mlr3_0.14.1      lubridate_1.9.2 </span></span>
<span><span class="co">##  [5] forcats_1.0.0    stringr_1.5.0    dplyr_1.1.0      purrr_1.0.1     </span></span>
<span><span class="co">##  [9] readr_2.1.4      tidyr_1.3.0      tibble_3.2.0     ggplot2_3.4.1   </span></span>
<span><span class="co">## [13] tidyverse_2.0.0  BiocStyle_2.26.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] uuid_1.1-0             backports_1.4.1        corrplot_0.92         </span></span>
<span><span class="co">##   [4] systemfonts_1.0.4      plyr_1.8.8             igraph_1.4.1          </span></span>
<span><span class="co">##   [7] splines_4.2.2          listenv_0.9.0          GenomeInfoDb_1.34.9   </span></span>
<span><span class="co">##  [10] gridBase_0.4-7         digest_0.6.31          foreach_1.5.2         </span></span>
<span><span class="co">##  [13] htmltools_0.5.4        lmerTest_3.1-3         fansi_1.0.4           </span></span>
<span><span class="co">##  [16] magrittr_2.0.3         checkmate_2.1.0        memoise_2.0.1         </span></span>
<span><span class="co">##  [19] cluster_2.1.4          tzdb_0.3.0             globals_0.16.2        </span></span>
<span><span class="co">##  [22] Biostrings_2.66.0      mlr3tuning_0.18.0      matrixStats_0.63.0    </span></span>
<span><span class="co">##  [25] vroom_1.6.1            timechange_0.2.0       pkgdown_2.0.7         </span></span>
<span><span class="co">##  [28] prettyunits_1.1.1      colorspace_2.1-0       textshaping_0.3.6     </span></span>
<span><span class="co">##  [31] xfun_0.37              crayon_1.5.2           RCurl_1.98-1.10       </span></span>
<span><span class="co">##  [34] jsonlite_1.8.4         lme4_1.1-32            survival_3.5-5        </span></span>
<span><span class="co">##  [37] iterators_1.0.14       ape_5.7-1              glue_1.6.2            </span></span>
<span><span class="co">##  [40] gtable_0.3.1           zlibbioc_1.44.0        XVector_0.38.0        </span></span>
<span><span class="co">##  [43] Rhdf5lib_1.20.0        shape_1.4.6            BiocGenerics_0.44.0   </span></span>
<span><span class="co">##  [46] scales_1.2.1           infotheo_1.2.0.1       DBI_1.1.3             </span></span>
<span><span class="co">##  [49] Rcpp_1.0.10            progress_1.2.2         palmerpenguins_0.1.1  </span></span>
<span><span class="co">##  [52] bit_4.0.5              stats4_4.2.2           glmnet_4.1-6          </span></span>
<span><span class="co">##  [55] RColorBrewer_1.1-3     ellipsis_0.3.2         farver_2.1.1          </span></span>
<span><span class="co">##  [58] pkgconfig_2.0.3        sass_0.4.5             utf8_1.2.3            </span></span>
<span><span class="co">##  [61] labeling_0.4.2         tidyselect_1.2.0       rlang_1.1.0           </span></span>
<span><span class="co">##  [64] reshape2_1.4.4         PRROC_1.3.1            munsell_0.5.0         </span></span>
<span><span class="co">##  [67] tools_4.2.2            cachem_1.0.7           cli_3.6.0             </span></span>
<span><span class="co">##  [70] generics_0.1.3         ade4_1.7-22            evaluate_0.20         </span></span>
<span><span class="co">##  [73] biomformat_1.26.0      fastmap_1.1.1          yaml_2.3.7            </span></span>
<span><span class="co">##  [76] ragg_1.2.5             bit64_4.0.5            knitr_1.42            </span></span>
<span><span class="co">##  [79] fs_1.6.1               lgr_0.4.4              beanplot_1.3.1        </span></span>
<span><span class="co">##  [82] bbotk_0.7.2            future_1.32.0          nlme_3.1-162          </span></span>
<span><span class="co">##  [85] paradox_0.11.0         compiler_4.2.2         rstudioapi_0.14       </span></span>
<span><span class="co">##  [88] curl_5.0.0             bslib_0.4.2            stringi_1.7.12        </span></span>
<span><span class="co">##  [91] highr_0.10             desc_1.4.2             lattice_0.20-45       </span></span>
<span><span class="co">##  [94] Matrix_1.5-3           nloptr_2.0.3           vegan_2.6-4           </span></span>
<span><span class="co">##  [97] permute_0.9-7          multtest_2.54.0        vctrs_0.5.2           </span></span>
<span><span class="co">## [100] pillar_1.8.1           lifecycle_1.0.3        rhdf5filters_1.10.0   </span></span>
<span><span class="co">## [103] BiocManager_1.30.20    jquerylib_0.1.4        LiblineaR_2.10-22     </span></span>
<span><span class="co">## [106] data.table_1.14.8      bitops_1.0-7           R6_2.5.1              </span></span>
<span><span class="co">## [109] bookdown_0.33          gridExtra_2.3          mlr3misc_0.11.0       </span></span>
<span><span class="co">## [112] IRanges_2.32.0         parallelly_1.34.0      codetools_0.2-19      </span></span>
<span><span class="co">## [115] boot_1.3-28.1          MASS_7.3-58.3          rhdf5_2.42.0          </span></span>
<span><span class="co">## [118] mlr3learners_0.5.6     rprojroot_2.0.3        withr_2.5.0           </span></span>
<span><span class="co">## [121] S4Vectors_0.36.2       GenomeInfoDbData_1.2.9 mgcv_1.8-42           </span></span>
<span><span class="co">## [124] parallel_4.2.2         hms_1.1.2              grid_4.2.2            </span></span>
<span><span class="co">## [127] minqa_1.2.5            rmarkdown_2.20         pROC_1.18.0           </span></span>
<span><span class="co">## [130] numDeriv_2016.8-1.1    Biobase_2.58.0</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Konrad Zych, Jakob Wirbel, Georg Zeller.</p>
</div>

<div class="privacy">
  <p><strong>Privacy:</strong> Usage of this site follows
      <a href="https://www.embl.de/aboutus/privacy_policy/index.html" class="external-link">EMBLâ€™s Privacy Policy</a><br>
      In accordance with that policy, we use Matomo to collect anonymised data on visits to, downloads from, and searches of this site.
      Contact <a href="mailto:zeller@embl.de">Georg Zeller</a> for details.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
