<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Holdout Testing with SIAMCAT • SIAMCAT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Holdout Testing with SIAMCAT">
<meta property="og:description" content="SIAMCAT">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><script type="text/javascript">
  var _paq = _paq || [];
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//tr-denbi.embl.de/heimdall/";
    _paq.push(['setTrackerUrl', u+'p.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'p.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SIAMCAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SIAMCAT_vignette.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/SIAMCAT_vignette.html">Introduction to SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_confounder.html">SIAMCAT example with confounders</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_holdout.html">Holdout testing with SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_meta.html">SIAMCAT machine learning meta-analysis</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_ml_pitfalls.html">Machine learning pitfalls</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_read-in.html">SIAMCAT input formats</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zellerlab/siamcat" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Holdout Testing with SIAMCAT</h1>
                        <h4 data-toc-skip class="author">Jakob Wirbel,
Konrad Zych, and Georg Zeller</h4>
            <address class="author_afil">
      EMBL
Heidelberg<br><a class="author_email" href="mailto:#"></a><a href="mailto:georg.zeller@embl.de" class="email">georg.zeller@embl.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">Date last modified:
2018-09-24</h4>
      
      
      <div class="hidden name"><code>SIAMCAT_holdout.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>One of the functionalities of the <code>SIAMCAT</code> package is the
training of statistical machine learning models on metagenomics data. In
this vignette, we demonstrate how such a model can be built on one
dataset and then be applied on another, similarly processed holdout
dataset. This might be of interest when comparing data from two
different studies on the same disease.</p>
<p>In this vignette, we look at two datasets from studies on colorectal
cancer (CRC). The first study from <a href="http://europepmc.org/abstract/MED/25432777" class="external-link">Zeller et al.</a>
investigated metagenomic markers for CRC in a population in France,
while the second study from <a href="https://europepmc.org/abstract/MED/26408641" class="external-link">Yu et al.</a> used
samples from China for the same goal. Both datasets were profiled with
the same <a href="https://github.com/motu-tool/mOTUs_v2" class="external-link">taxonomic
profiling tool</a>, yielding the same taxonomic identifiers, which is
required for holdout testing.</p>
</div>
<div class="section level2">
<h2 id="load-the-data">Load the Data<a class="anchor" aria-label="anchor" href="#load-the-data"></a>
</h2>
<p>The datasets can be found on the web repository for public
metagenomics datasets from the <a href="https://www.embl.de/research/units/scb/zeller/index.html" class="external-link">Zeller
group</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SIAMCAT"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data.loc</span> <span class="op">&lt;-</span> <span class="st">'https://zenodo.org/api/files/d81e429c-870f-44e0-a44a-2a4aa541b6c1/'</span></span>
<span><span class="co"># this is data from Zeller et al., Mol. Syst. Biol. 2014</span></span>
<span><span class="va">fn.meta.fr</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data.loc</span>, <span class="st">'meta_Zeller.tsv'</span><span class="op">)</span></span>
<span><span class="va">fn.feat.fr</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data.loc</span>, <span class="st">'specI_Zeller.tsv'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># this is the external dataset from Yu et al., Gut 2017</span></span>
<span><span class="va">fn.meta.cn</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data.loc</span>, <span class="st">'meta_Yu.tsv'</span><span class="op">)</span></span>
<span><span class="va">fn.feat.cn</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data.loc</span>, <span class="st">'specI_Yu.tsv'</span><span class="op">)</span></span></code></pre></div>
<p>First of all, we build a <code>SIAMCAT</code> object using the data
from the French study in the same way that we have seen before in the
main <code>SIAMCAT</code> vignette.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># features</span></span>
<span><span class="co"># be vary of the defaults in R!!!</span></span>
<span><span class="va">feat.fr</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">fn.feat.fr</span>, sep<span class="op">=</span><span class="st">'\t'</span>, quote<span class="op">=</span><span class="st">""</span>,</span>
<span>    check.names <span class="op">=</span> <span class="cn">FALSE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># the features are counts, but we want to work with relative abundances</span></span>
<span><span class="va">feat.fr.rel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/proportions.html" class="external-link">prop.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">feat.fr</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># metadata</span></span>
<span><span class="va">meta.fr</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">fn.meta.fr</span>, sep<span class="op">=</span><span class="st">'\t'</span>, quote<span class="op">=</span><span class="st">""</span>,</span>
<span>    check.names<span class="op">=</span><span class="cn">FALSE</span>, stringsAsFactors<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create SIAMCAT object</span></span>
<span><span class="va">siamcat.fr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.fr.rel</span>, meta<span class="op">=</span><span class="va">meta.fr</span>,</span>
<span>    label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'CRC'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## + starting create.label</span></span></code></pre>
<pre><code><span><span class="co">## Label used as case:</span></span>
<span><span class="co">##    CRC</span></span>
<span><span class="co">## Label used as control:</span></span>
<span><span class="co">##    CTR</span></span></code></pre>
<pre><code><span><span class="co">## + finished create.label.from.metadata in 0.021 s</span></span></code></pre>
<pre><code><span><span class="co">## + starting validate.data</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking overlap between labels and features</span></span></code></pre>
<pre><code><span><span class="co">## + Keeping labels of 141 sample(s).</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking sample number per class</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking overlap between samples and metadata</span></span></code></pre>
<pre><code><span><span class="co">## + finished validate.data in 0.039 s</span></span></code></pre>
<p>We can load the data from the Chinese study in a similar way and also
create a <code>SIAMCAT</code> object for the holdout dataset.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># features</span></span>
<span><span class="va">feat.cn</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">fn.feat.cn</span>, sep<span class="op">=</span><span class="st">'\t'</span>, quote<span class="op">=</span><span class="st">""</span>,</span>
<span>    check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">feat.cn.rel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/proportions.html" class="external-link">prop.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">feat.cn</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># metadata</span></span>
<span><span class="va">meta.cn</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">fn.meta.cn</span>, sep<span class="op">=</span><span class="st">'\t'</span>, quote<span class="op">=</span><span class="st">""</span>,</span>
<span>    check.names<span class="op">=</span><span class="cn">FALSE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># SIAMCAT object</span></span>
<span><span class="va">siamcat.cn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.cn.rel</span>, meta<span class="op">=</span><span class="va">meta.cn</span>,</span>
<span>        label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'CRC'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## + starting create.label</span></span></code></pre>
<pre><code><span><span class="co">## Label used as case:</span></span>
<span><span class="co">##    CRC</span></span>
<span><span class="co">## Label used as control:</span></span>
<span><span class="co">##    CTR</span></span></code></pre>
<pre><code><span><span class="co">## + finished create.label.from.metadata in 0.001 s</span></span></code></pre>
<pre><code><span><span class="co">## + starting validate.data</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking overlap between labels and features</span></span></code></pre>
<pre><code><span><span class="co">## + Keeping labels of 128 sample(s).</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking sample number per class</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking overlap between samples and metadata</span></span></code></pre>
<pre><code><span><span class="co">## + finished validate.data in 0.025 s</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="model-building-on-the-french-dataset">Model Building on the French Dataset<a class="anchor" aria-label="anchor" href="#model-building-on-the-french-dataset"></a>
</h2>
<div class="section level3">
<h3 id="preprocessing">Preprocessing<a class="anchor" aria-label="anchor" href="#preprocessing"></a>
</h3>
<p>With the French dataset, we perform the complete process of model
building within <code>SIAMCAT</code>, including data preprocessing steps
like data validation, filtering, and data normalization.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat.fr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span></span>
<span>    <span class="va">siamcat.fr</span>,</span>
<span>    filter.method <span class="op">=</span> <span class="st">'abundance'</span>,</span>
<span>    cutoff <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>    rm.unmapped <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    verbose<span class="op">=</span><span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## + starting filter.features</span></span></code></pre>
<pre><code><span><span class="co">## +++ before filtering, the data have 1754 features</span></span></code></pre>
<pre><code><span><span class="co">## +++ removed 1 features corresponding to UNMAPPED reads</span></span></code></pre>
<pre><code><span><span class="co">## +++ removed 1539 features whose values did not exceed 0.001 in any sample (retaining 215)</span></span></code></pre>
<pre><code><span><span class="co">## + finished filter.features in 0.003 s</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat.fr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span></span>
<span>    <span class="va">siamcat.fr</span>,</span>
<span>    norm.method <span class="op">=</span> <span class="st">"log.std"</span>,</span>
<span>    norm.param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>log.n0 <span class="op">=</span> <span class="fl">1e-06</span>, sd.min.q <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>,</span>
<span>    verbose <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## + starting normalize.features</span></span></code></pre>
<pre><code><span><span class="co">## +++ performing de novo normalization using the  log.std  method</span></span></code></pre>
<pre><code><span><span class="co">## + feature sparsity before normalization: 46.05%</span></span></code></pre>
<pre><code><span><span class="co">## +++ feature sparsity after normalization:      0 %</span></span></code></pre>
<pre><code><span><span class="co">## + finished normalize.features in 0.005 s</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="model-training">Model Training<a class="anchor" aria-label="anchor" href="#model-training"></a>
</h3>
<p>Now, we can build the statistical model. We use the same parameters
as in the main <code>SIAMCAT</code> vignette, where the process is
explained in more detail.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat.fr</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span></span>
<span>    <span class="va">siamcat.fr</span>,</span>
<span>    num.folds <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    num.resample <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Features splitted for cross-validation successfully.</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat.fr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span></span>
<span>    <span class="va">siamcat.fr</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"lasso"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Trained lasso models successfully.</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="predictions">Predictions<a class="anchor" aria-label="anchor" href="#predictions"></a>
</h3>
<p>Finally, we can make predictions for each cross-validation fold and
evaluate the predictions as seen in the main <code>SIAMCAT</code>
vignette.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat.fr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">siamcat.fr</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Made predictions successfully.</span></span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat.fr</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">siamcat.fr</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Evaluated predictions successfully.</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="application-on-the-holdout-dataset">Application on the Holdout Dataset<a class="anchor" aria-label="anchor" href="#application-on-the-holdout-dataset"></a>
</h2>
<p>Now that we have successfully built the model for the French dataset,
we can apply it to the Chinese holdout dataset. First, we will normalize
the Chinese dataset with the same parameters that we used for the French
dataset in order to make the data comparable. For that step, we can use
the frozen normalization functionality in the
<code>normalize.features</code> function in <code>SIAMCAT</code>. We
supply to the function all normalization parameters saved in the
<code>siamcat.fr</code> object, which can be accessed using the
<code>norm_params</code> accessor.</p>
<div class="section level3">
<h3 id="frozen-normalization">Frozen Normalization<a class="anchor" aria-label="anchor" href="#frozen-normalization"></a>
</h3>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat.cn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">siamcat.cn</span>,</span>
<span>    norm.param<span class="op">=</span><span class="fu"><a href="../reference/norm_params-methods.html">norm_params</a></span><span class="op">(</span><span class="va">siamcat.fr</span><span class="op">)</span>,</span>
<span>    feature.type<span class="op">=</span><span class="st">'original'</span>,</span>
<span>    verbose <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## + starting normalize.features</span></span></code></pre>
<pre><code><span><span class="co">## + normalizing original features</span></span></code></pre>
<pre><code><span><span class="co">## + performing frozen  log.std  normalization using the supplied parameters</span></span></code></pre>
<pre><code><span><span class="co">## + feature sparsity before normalization: 49.77%</span></span></code></pre>
<pre><code><span><span class="co">## + feature sparsity after normalization:     0%</span></span></code></pre>
<pre><code><span><span class="co">## + finished normalize.features in 0.002 s</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="holdout-predictions">Holdout Predictions<a class="anchor" aria-label="anchor" href="#holdout-predictions"></a>
</h3>
<p>Next, we apply the trained model to predict the holdout dataset.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat.cn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span></span>
<span>    siamcat <span class="op">=</span> <span class="va">siamcat.fr</span>,</span>
<span>    siamcat.holdout <span class="op">=</span> <span class="va">siamcat.cn</span>,</span>
<span>    normalize.holdout <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in make.external.predictions(siamcat.trained = siamcat,</span></span>
<span><span class="co">## siamcat.external = siamcat.holdout, : WARNING: holdout set is not being</span></span>
<span><span class="co">## normalized!</span></span></code></pre>
<pre><code><span><span class="co">## Made predictions successfully.</span></span></code></pre>
<p>Note that the <code>make.predictions</code> function can also take
care of the normalization of the holdout dataset.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Alternative Code, not run here</span></span>
<span><span class="va">siamcat.cn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.cn.rel</span>, meta<span class="op">=</span><span class="va">meta.cn</span>,</span>
<span>    label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'CRC'</span><span class="op">)</span></span>
<span><span class="va">siamcat.cn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span>siamcat <span class="op">=</span> <span class="va">siamcat.fr</span>,</span>
<span>    siamcat.holdout <span class="op">=</span> <span class="va">siamcat.cn</span>,</span>
<span>    normalize.holdout <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Again, we have to evaluate the predictions:</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat.cn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">siamcat.cn</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="model-evaluation">Model Evaluation<a class="anchor" aria-label="anchor" href="#model-evaluation"></a>
</h2>
<p>Now, we can compare the performance of the classifier on the original
and the holdout dataset by using the <code>model.evaluation.plot</code>
function. Here, we can supply several <code>SIAMCAT</code> objects for
which the model evaluation will be plotted in the same plot. Note that
we can supply the objects as named objects in order to print the names
in the legend.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/model.evaluation.plot.html">model.evaluation.plot</a></span><span class="op">(</span><span class="st">'FR-CRC'</span><span class="op">=</span><span class="va">siamcat.fr</span>,</span>
<span>    <span class="st">'CN-CRC'</span><span class="op">=</span><span class="va">siamcat.cn</span>,</span>
<span>    colours<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'dimgrey'</span>, <span class="st">'orange'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="eval_plot_holdout.png"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] SIAMCAT_2.3.3    phyloseq_1.42.0  mlr3_0.14.1      BiocStyle_2.26.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] paradox_0.11.0         minqa_1.2.5            colorspace_2.1-0      </span></span>
<span><span class="co">##   [4] ellipsis_0.3.2         rprojroot_2.0.3        XVector_0.38.0        </span></span>
<span><span class="co">##   [7] fs_1.6.1               rstudioapi_0.14        listenv_0.9.0         </span></span>
<span><span class="co">##  [10] mlr3tuning_0.18.0      fansi_1.0.4            codetools_0.2-19      </span></span>
<span><span class="co">##  [13] splines_4.2.2          mlr3learners_0.5.6     PRROC_1.3.1           </span></span>
<span><span class="co">##  [16] cachem_1.0.7           knitr_1.42             ade4_1.7-22           </span></span>
<span><span class="co">##  [19] jsonlite_1.8.4         nloptr_2.0.3           pROC_1.18.0           </span></span>
<span><span class="co">##  [22] gridBase_0.4-7         cluster_2.1.4          BiocManager_1.30.20   </span></span>
<span><span class="co">##  [25] compiler_4.2.2         backports_1.4.1        Matrix_1.5-3          </span></span>
<span><span class="co">##  [28] fastmap_1.1.1          cli_3.6.0              prettyunits_1.1.1     </span></span>
<span><span class="co">##  [31] htmltools_0.5.4        tools_4.2.2            lmerTest_3.1-3        </span></span>
<span><span class="co">##  [34] igraph_1.4.1           gtable_0.3.1           glue_1.6.2            </span></span>
<span><span class="co">##  [37] GenomeInfoDbData_1.2.9 reshape2_1.4.4         LiblineaR_2.10-22     </span></span>
<span><span class="co">##  [40] dplyr_1.1.0            Rcpp_1.0.10            Biobase_2.58.0        </span></span>
<span><span class="co">##  [43] jquerylib_0.1.4        pkgdown_2.0.7          vctrs_0.5.2           </span></span>
<span><span class="co">##  [46] Biostrings_2.66.0      rhdf5filters_1.10.0    multtest_2.54.0       </span></span>
<span><span class="co">##  [49] ape_5.7-1              nlme_3.1-162           iterators_1.0.14      </span></span>
<span><span class="co">##  [52] xfun_0.37              stringr_1.5.0          mlr3measures_0.5.0    </span></span>
<span><span class="co">##  [55] globals_0.16.2         lme4_1.1-32            lifecycle_1.0.3       </span></span>
<span><span class="co">##  [58] beanplot_1.3.1         future_1.32.0          zlibbioc_1.44.0       </span></span>
<span><span class="co">##  [61] MASS_7.3-58.3          scales_1.2.1           lgr_0.4.4             </span></span>
<span><span class="co">##  [64] hms_1.1.2              ragg_1.2.5             parallel_4.2.2        </span></span>
<span><span class="co">##  [67] biomformat_1.26.0      rhdf5_2.42.0           RColorBrewer_1.1-3    </span></span>
<span><span class="co">##  [70] yaml_2.3.7             memoise_2.0.1          gridExtra_2.3         </span></span>
<span><span class="co">##  [73] ggplot2_3.4.1          sass_0.4.5             stringi_1.7.12        </span></span>
<span><span class="co">##  [76] S4Vectors_0.36.2       desc_1.4.2             corrplot_0.92         </span></span>
<span><span class="co">##  [79] foreach_1.5.2          checkmate_2.1.0        permute_0.9-7         </span></span>
<span><span class="co">##  [82] palmerpenguins_0.1.1   BiocGenerics_0.44.0    boot_1.3-28.1         </span></span>
<span><span class="co">##  [85] shape_1.4.6            GenomeInfoDb_1.34.9    matrixStats_0.63.0    </span></span>
<span><span class="co">##  [88] rlang_1.1.0            pkgconfig_2.0.3        systemfonts_1.0.4     </span></span>
<span><span class="co">##  [91] bitops_1.0-7           evaluate_0.20          lattice_0.20-45       </span></span>
<span><span class="co">##  [94] purrr_1.0.1            Rhdf5lib_1.20.0        tidyselect_1.2.0      </span></span>
<span><span class="co">##  [97] parallelly_1.34.0      plyr_1.8.8             magrittr_2.0.3        </span></span>
<span><span class="co">## [100] bookdown_0.33          R6_2.5.1               IRanges_2.32.0        </span></span>
<span><span class="co">## [103] generics_0.1.3         DBI_1.1.3              pillar_1.8.1          </span></span>
<span><span class="co">## [106] mgcv_1.8-42            survival_3.5-5         RCurl_1.98-1.10       </span></span>
<span><span class="co">## [109] tibble_3.2.0           crayon_1.5.2           uuid_1.1-0            </span></span>
<span><span class="co">## [112] utf8_1.2.3             rmarkdown_2.20         progress_1.2.2        </span></span>
<span><span class="co">## [115] grid_4.2.2             data.table_1.14.8      vegan_2.6-4           </span></span>
<span><span class="co">## [118] infotheo_1.2.0.1       mlr3misc_0.11.0        bbotk_0.7.2           </span></span>
<span><span class="co">## [121] digest_0.6.31          numDeriv_2016.8-1.1    textshaping_0.3.6     </span></span>
<span><span class="co">## [124] stats4_4.2.2           munsell_0.5.0          glmnet_4.1-6          </span></span>
<span><span class="co">## [127] bslib_0.4.2</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Konrad Zych, Jakob Wirbel, Georg Zeller.</p>
</div>

<div class="privacy">
  <p><strong>Privacy:</strong> Usage of this site follows
      <a href="https://www.embl.de/aboutus/privacy_policy/index.html" class="external-link">EMBL’s Privacy Policy</a><br>
      In accordance with that policy, we use Matomo to collect anonymised data on visits to, downloads from, and searches of this site.
      Contact <a href="mailto:zeller@embl.de">Georg Zeller</a> for details.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
