<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Example dataset with Confoundering • SIAMCAT</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Example dataset with Confoundering">
<meta property="og:description" content="SIAMCAT">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><script type="text/javascript">
  var _paq = _paq || [];
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//tr-denbi.embl.de/heimdall/";
    _paq.push(['setTrackerUrl', u+'p.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'p.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SIAMCAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.11.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SIAMCAT_vignette.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/SIAMCAT_vignette.html">Introduction to SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_confounder.html">SIAMCAT example with confounders</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_holdout.html">Holdout testing with SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_meta.html">SIAMCAT machine learning meta-analysis</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_ml_pitfalls.html">Machine learning pitfalls</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_read-in.html">SIAMCAT input formats</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://twitter.com/siamcat_dev">
    <span class="fas fa-twitter"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/zellerlab/siamcat">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="SIAMCAT_confounder_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Example dataset with Confoundering</h1>
                        <h4 data-toc-skip class="author">Jakob Wirbel and Georg Zeller</h4>
            <address class="author_afil">
      EMBL Heidelberg<br><a class="author_email" href="mailto:#"></a><a href="mailto:georg.zeller@embl.de" class="email">georg.zeller@embl.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">Date last modified: 2020-11-11</h4>
      
      
      <div class="hidden name"><code>SIAMCAT_confounder.Rmd</code></div>

    </div>

    
    
<div id="about-this-vignette" class="section level1">
<h1 class="hasAnchor">
<a href="#about-this-vignette" class="anchor"></a>About This Vignette</h1>
<p>Here, we demonstrate the standard workflow of the <code>SIAMCAT</code> package using as an example the dataset from <a href="https://www.ncbi.nlm.nih.gov/pubmed/24997787">Nielsen et al. <em>Nat Biotechnol</em> 2014</a>. This dataset contains samples from patients with inflammatory bowel disease and from controls.<br>
More importantly, these samples have been collected in two different countries, Spain and Denmark. Together with technical differences between these samples, this introduces a potent confounding factor into the data. Here we are going to explore how <code>SIAMCAT</code> would identify the confounding factor and what the results would be if you account for the confounder or not.</p>
<div id="setup" class="section level2">
<h2 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h2>
<p>First, we load the packages needed to perform the analyses.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://tidyverse.tidyverse.org">"tidyverse"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"SIAMCAT"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://rpkgs.datanovia.com/ggpubr/">"ggpubr"</a></span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="preparations" class="section level1">
<h1 class="hasAnchor">
<a href="#preparations" class="anchor"></a>Preparations</h1>
<p>There are two different ways to access the data for our example dataset. On the one hand, it is available through the <code>curatedMetagenomicsData</code> R package. However, using it here would create many more dependencies for the <code>SIAMCAT</code> package.<br>
Therefore, we here use data available through the EMBL cluster.</p>
<p>In the <code>SIAMCAT</code> paper, we performed the presented analyses on the datasets available through <code>curatedMetagenomicsData</code>. If you want to reproduce the analysis from the <code>SIAMCAT</code> paper, you can execute the code chunks in the <code>curatedMetageomicsData</code> section, otherwise execute the code in the mOTUs2 section.</p>
<div id="curatedmetagenomicsdata" class="section level2">
<h2 class="hasAnchor">
<a href="#curatedmetagenomicsdata" class="anchor"></a>curatedMetagenomicsData</h2>
<p>First, we load the package:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"curatedMetagenomciData"</span><span class="op">)</span></code></pre></div>
<div id="metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#metadata" class="anchor"></a>Metadata</h3>
<p>The data are part of the <code>combined_metadata</code></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta.nielsen.full</span> <span class="op">&lt;-</span> <span class="va">combined_metadata</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">dataset_name</span><span class="op">==</span><span class="st">'NielsenHB_2014'</span><span class="op">)</span></code></pre></div>
<p>One thing we have to keep in mind are repeated samples per subject (see also the <strong>Machine learning pitfalls</strong> vignette).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">meta.nielsen.full</span><span class="op">$</span><span class="va">subjectID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">meta.nielsen.full</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Some subjects (but not all) had been sampled multiple times. Therefore, we want to remove repeated samplings for the same subject, since the samples would otherwise not be indepdenent from another.</p>
<p>The visit number is encoded in the <code>sampleID</code>. Therefore, we can use this information to extract when the samples have been taken and use only the first visit for each subject.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen.full</span> <span class="op">%&gt;%</span>
  <span class="fu">select</span><span class="op">(</span><span class="va">sampleID</span>, <span class="va">subjectID</span>, <span class="va">study_condition</span>, <span class="va">disease_subtype</span>,
         <span class="va">disease</span>, <span class="va">age</span>, <span class="va">country</span>, <span class="va">number_reads</span>, <span class="va">median_read_length</span>, <span class="va">BMI</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>visit<span class="op">=</span><span class="fu">str_extract</span><span class="op">(</span><span class="va">sampleID</span>, <span class="st">'_[0-9]+$'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>visit<span class="op">=</span><span class="fu">str_remove</span><span class="op">(</span><span class="va">visit</span>, <span class="st">'_'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>visit<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">visit</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>visit<span class="op">=</span><span class="fu">case_when</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">visit</span><span class="op">)</span><span class="op">~</span><span class="fl">0</span>, <span class="cn">TRUE</span><span class="op">~</span><span class="va">visit</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">subjectID</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">visit</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">visit</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Sample_ID<span class="op">=</span><span class="va">sampleID</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Group<span class="op">=</span><span class="fu">case_when</span><span class="op">(</span><span class="va">disease</span><span class="op">==</span><span class="st">'healthy'</span><span class="op">~</span><span class="st">'CTR'</span>,
                         <span class="cn">TRUE</span><span class="op">~</span><span class="va">disease_subtype</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Now, we can restrict our metadata to samples with <code>UC</code> and healthy control samples:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Group</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'UC'</span>, <span class="st">'CTR'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>As a last step, we can adjust the column names for the metadata so that they agree with the data available from the EMBL cluster. Also, we add rownames to the dataframe since <code>SIAMCAT</code> needs rownames to match samples across metadata and features.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Country<span class="op">=</span><span class="va">country</span><span class="op">)</span>
<span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">meta.nielsen</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">meta.nielsen</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span><span class="op">$</span><span class="va">sampleID</span></code></pre></div>
</div>
<div id="taxonomic-profiles" class="section level3">
<h3 class="hasAnchor">
<a href="#taxonomic-profiles" class="anchor"></a>Taxonomic Profiles</h3>
<p>We can load the taxonomic profiles generated with MetaPhlAn2 via the <em>curatedMetagenomicsData</em> R package.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="st">'NielsenHB_2014.metaphlan_bugs_list.stool'</span>
<span class="va">feat</span> <span class="op">&lt;-</span> <span class="fu">curatedMetagenomicData</span><span class="op">(</span>x<span class="op">=</span><span class="va">x</span>, dryrun<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="va">feat</span> <span class="op">&lt;-</span> <span class="va">feat</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">assayData</span><span class="op">$</span><span class="va">exprs</span></code></pre></div>
<p>The MetaPhlAn2 profiles contain information on different taxonomic levels. Therefore, we want to restrict them to species-level profiles. In a second step, we convert them into relative abundances (summing up to 1) instead of using the percentages (summing up to 100) that MetaPhlAn2 outputs.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">feat</span> <span class="op">&lt;-</span> <span class="va">feat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span>, pattern<span class="op">=</span><span class="st">'s__'</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">feat</span> <span class="op">&lt;-</span> <span class="va">feat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span>,pattern<span class="op">=</span><span class="st">'t__'</span>, invert <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">feat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span><span class="op">/</span><span class="fl">100</span><span class="op">)</span></code></pre></div>
<p>The feature names are very long and may be a bit un-wieldy for plotting later on, so we shorten them to only the species name:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">str_extract</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span>, <span class="st">'s__.*$'</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="motus2-profiles" class="section level2">
<h2 class="hasAnchor">
<a href="#motus2-profiles" class="anchor"></a>mOTUs2 Profiles</h2>
<p>Both metadata and features are available through the EMBL cluster:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># base url for data download</span>
<span class="va">data.location</span> <span class="op">&lt;-</span> <span class="st">'https://www.embl.de/download/zeller/metaHIT/'</span>
<span class="co">## metadata</span>
<span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">data.location</span>, <span class="st">'meta_Nielsen.tsv'</span><span class="op">)</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## ── Column specification ────────────────────────────────────────────────────────</span>
<span class="co">## cols(</span>
<span class="co">##   Sample_ID = col_character(),</span>
<span class="co">##   Individual_ID = col_character(),</span>
<span class="co">##   Country = col_character(),</span>
<span class="co">##   Sampling_day = col_double(),</span>
<span class="co">##   Gender = col_character(),</span>
<span class="co">##   Age = col_double(),</span>
<span class="co">##   BMI = col_double(),</span>
<span class="co">##   Group = col_character(),</span>
<span class="co">##   Library_Size = col_double()</span>
<span class="co">## )</span>
<span class="co"># also here, we have to remove repeated samplings and CD samples</span>
<span class="va">meta.nielsen</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Group</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'CTR'</span>, <span class="st">'UC'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Individual_ID</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Sampling_day</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">Sampling_day</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">meta.nielsen</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span><span class="op">$</span><span class="va">Sample_ID</span>

<span class="co">## features</span>
<span class="va">feat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">data.location</span>, <span class="st">'metaHIT_motus.tsv'</span><span class="op">)</span>, 
                   stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>, sep<span class="op">=</span><span class="st">'\t'</span>,
                   check.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="st">''</span>, comment.char <span class="op">=</span> <span class="st">''</span><span class="op">)</span>
<span class="va">feat</span> <span class="op">&lt;-</span> <span class="va">feat</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span>
<span class="va">feat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/proportions.html">prop.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">feat</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="siamcat-workflow-without-confounders" class="section level1">
<h1 class="hasAnchor">
<a href="#siamcat-workflow-without-confounders" class="anchor"></a>SIAMCAT Workflow (without Confounders)</h1>
<div id="the-siamcat-object" class="section level2">
<h2 class="hasAnchor">
<a href="#the-siamcat-object" class="anchor"></a>The SIAMCAT Object</h2>
<p>Now, we have everything ready to create a <code>SIAMCAT</code> object which stores the feature matrix, the meta-variables, and the label. Here, the label is created using the information in the metadata.<br>
To demonstrate the normal <code>SIAMCAT</code> workflow, we remove the confounding factor by only looking at samples from Spain. Below, we have a look what would have happened if we had not removed them.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># remove Danish samples</span>
<span class="va">meta.nielsen.esp</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span><span class="op">[</span><span class="va">meta.nielsen</span><span class="op">$</span><span class="va">Country</span> <span class="op">==</span> <span class="st">'ESP'</span>,<span class="op">]</span>
<span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat</span>, meta<span class="op">=</span><span class="va">meta.nielsen.esp</span>, label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'UC'</span><span class="op">)</span>
<span class="co">## + starting create.label</span>
<span class="co">## Label used as case:</span>
<span class="co">##    UC</span>
<span class="co">## Label used as control:</span>
<span class="co">##    CTR</span>
<span class="co">## + finished create.label.from.metadata in 0.02 s</span>
<span class="co">## + starting validate.data</span>
<span class="co">## +++ checking overlap between labels and features</span>
<span class="co">## + Keeping labels of 128 sample(s).</span>
<span class="co">## +++ checking sample number per class</span>
<span class="co">## +++ checking overlap between samples and metadata</span>
<span class="co">## + finished validate.data in 0.377 s</span></code></pre></div>
</div>
<div id="filtering" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering" class="anchor"></a>Filtering</h2>
<p>Now, we can filter feature with low overall abundance and prevalence.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj</span>, cutoff<span class="op">=</span><span class="fl">1e-04</span>,
                          filter.method <span class="op">=</span> <span class="st">'abundance'</span><span class="op">)</span>
<span class="co">## Features successfully filtered</span>
<span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj</span>, cutoff<span class="op">=</span><span class="fl">0.05</span>,
                          filter.method<span class="op">=</span><span class="st">'prevalence'</span>,
                          feature.type <span class="op">=</span> <span class="st">'filtered'</span><span class="op">)</span>
<span class="co">## Features successfully filtered</span></code></pre></div>
</div>
<div id="association-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#association-plot" class="anchor"></a>Association Plot</h2>
<p>The <code>check.assocation</code> function calculates the significance of enrichment and metrics of association (such as generalized fold change and single-feautre AUROC).</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/check.associations.html">check.associations</a></span><span class="op">(</span><span class="va">sc.obj</span>, detect.lim <span class="op">=</span> <span class="fl">1e-06</span>,
                             alpha<span class="op">=</span><span class="fl">0.1</span>, max.show <span class="op">=</span> <span class="fl">20</span>,
                             plot.type <span class="op">=</span> <span class="st">'quantile.rect'</span>,
                             panels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'fc'</span><span class="op">)</span>,
                             fn.plot <span class="op">=</span> <span class="st">'./association_plot_nielsen.pdf'</span><span class="op">)</span></code></pre></div>
<p><img src="association_plot_nielsen.png"></p>
</div>
<div id="confounder-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#confounder-analysis" class="anchor"></a>Confounder Analysis</h2>
<p>We can also check the supplied meta-variables for potential confounding.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/check.confounders.html">check.confounders</a></span><span class="op">(</span><span class="va">sc.obj</span>, fn.plot <span class="op">=</span> <span class="st">'./confounders_nielsen.pdf'</span><span class="op">)</span>
<span class="co">## ++ remove metadata variables, since all subjects have the same value</span>
<span class="co">##  Country</span>
<span class="co">## Finished checking metadata for confounders, results plotted to: ./confounders_nielsen.pdf</span></code></pre></div>
<p><img src="confounders_nielsen.png"></p>
<p>The function produces one plot for each meta-variable. Here, we show only the example of the body mass index (BMI). The BMI distributions look very similar for both controls and UC cases, so it is unlikely that the BMI would confound the analyses.</p>
</div>
<div id="machine-learning-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#machine-learning-workflow" class="anchor"></a>Machine Learning Workflow</h2>
<p>The machine learning workflow can be easily implemented in <code>SIAMCAT</code>. It contains the following steps:</p>
<ul>
<li>Feature normalization</li>
<li>Data splitting for cross-validation</li>
<li>Model training</li>
<li>Making model predictions (on left-out data)</li>
<li>Evaluating model predictions (using AUROC and AUPRC)</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">sc.obj</span>, norm.method <span class="op">=</span> <span class="st">'log.std'</span>,
                             norm.param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>log.n0<span class="op">=</span><span class="fl">1e-06</span>, sd.min.q<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span>
<span class="co">## Features normalized successfully.</span>
<span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span><span class="va">sc.obj</span>, num.folds <span class="op">=</span> <span class="fl">5</span>, num.resample <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="co">## Features splitted for cross-validation successfully.</span>
<span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj</span>, method<span class="op">=</span><span class="st">'lasso'</span><span class="op">)</span>
<span class="co">## Trained lasso models successfully.</span>
<span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span>
<span class="co">## Made predictions successfully.</span>
<span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span>
<span class="co">## Evaluated predictions successfully.</span></code></pre></div>
<div id="model-evaluation-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#model-evaluation-plot" class="anchor"></a>Model Evaluation Plot</h3>
<p>The model evaluation plot will produce one plot with the ROC curve and another one with the precision-recall curve (not shown here).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/model.evaluation.plot.html">model.evaluation.plot</a></span><span class="op">(</span><span class="va">sc.obj</span>, fn.plot <span class="op">=</span> <span class="st">'./eval_plot_nielsen.pdf'</span><span class="op">)</span>
<span class="co">## Plotted evaluation of predictions successfully to: ./eval_plot_nielsen.pdf</span></code></pre></div>
<p><img src="eval_plot_nielsen.png"></p>
</div>
<div id="model-interpretation-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#model-interpretation-plot" class="anchor"></a>Model Interpretation Plot</h3>
<p>The model interpretation plot can give you additional information about the trained machine learning model. It will show you:</p>
<ul>
<li>the feature importance as barplot,</li>
<li>the feature robustness (in how many of the models in the repeated cross-validation this feature has been selected into the model),</li>
<li>the normalized feature abundances across samples as heatmap,</li>
<li>the optional metadata as heatmap below, and</li>
<li>a boxplot showing the proportion of the model weight that is explained by<br>
the selected features.</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/model.interpretation.plot.html">model.interpretation.plot</a></span><span class="op">(</span><span class="va">sc.obj</span>, consens.thres <span class="op">=</span> <span class="fl">0.8</span>,
                          fn.plot <span class="op">=</span> <span class="st">'./interpretation_nielsen.pdf'</span><span class="op">)</span>
<span class="co">## Successfully plotted model interpretation plot to: ./interpretation_nielsen.pdf</span></code></pre></div>
<p><img src="interpretation_nielsen.png"></p>
</div>
</div>
</div>
<div id="confounder-analysis-1" class="section level1">
<h1 class="hasAnchor">
<a href="#confounder-analysis-1" class="anchor"></a>Confounder Analysis</h1>
<p>As already mentioned above, the Nielsen dataset contains samples from both Spain and Denmark. How would <code>SIAMCAT</code> have alerted us to this?</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">meta.nielsen</span><span class="op">$</span><span class="va">Group</span>, <span class="va">meta.nielsen</span><span class="op">$</span><span class="va">Country</span><span class="op">)</span>
<span class="co">##      </span>
<span class="co">##       DNK ESP</span>
<span class="co">##   CTR 177  59</span>
<span class="co">##   UC    0  69</span></code></pre></div>
<div id="country-confounder" class="section level2">
<h2 class="hasAnchor">
<a href="#country-confounder" class="anchor"></a>Country Confounder</h2>
<p>First, we create a <code>SIAMCAT</code> object again, this time including the Danish controls:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat</span>, meta<span class="op">=</span><span class="va">meta.nielsen</span>,
                       label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'UC'</span><span class="op">)</span>
<span class="co">## + starting create.label</span>
<span class="co">## Label used as case:</span>
<span class="co">##    UC</span>
<span class="co">## Label used as control:</span>
<span class="co">##    CTR</span>
<span class="co">## + finished create.label.from.metadata in 0.001 s</span>
<span class="co">## + starting validate.data</span>
<span class="co">## +++ checking overlap between labels and features</span>
<span class="co">## + Keeping labels of 291 sample(s).</span>
<span class="co">## + Removed 14 samples from the label object...</span>
<span class="co">## +++ checking sample number per class</span>
<span class="co">## +++ checking overlap between samples and metadata</span>
<span class="co">## + finished validate.data in 0.463 s</span>
<span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, cutoff<span class="op">=</span><span class="fl">1e-04</span>,
                               filter.method <span class="op">=</span> <span class="st">'abundance'</span><span class="op">)</span>
<span class="co">## Features successfully filtered</span>
<span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, cutoff<span class="op">=</span><span class="fl">0.05</span>,
                               filter.method<span class="op">=</span><span class="st">'prevalence'</span>,
                               feature.type <span class="op">=</span> <span class="st">'filtered'</span><span class="op">)</span>
<span class="co">## Features successfully filtered</span></code></pre></div>
<p>The confounder plot would show us that the meta-variable “country” might be problematic:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/check.confounders.html">check.confounders</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, fn.plot <span class="op">=</span> <span class="st">'./confounders_dnk.pdf'</span><span class="op">)</span></code></pre></div>
<p><img src="confounders_dnk.png"></p>
</div>
<div id="association-testing" class="section level2">
<h2 class="hasAnchor">
<a href="#association-testing" class="anchor"></a>Association Testing</h2>
<p>First, we can use <code>SIAMCAT</code> to test for associations including the Danish samples.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/check.associations.html">check.associations</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, detect.lim <span class="op">=</span> <span class="fl">1e-06</span>,
                                  alpha<span class="op">=</span><span class="fl">0.1</span>, max.show <span class="op">=</span> <span class="fl">20</span>,
                                  plot.type <span class="op">=</span> <span class="st">'quantile.rect'</span>,
                                  fn.plot <span class="op">=</span> <span class="st">'./association_plot_dnk.pdf'</span><span class="op">)</span></code></pre></div>
<p>Confounders can lead to biases in association testing. After using <code>SIAMCAT</code> to test for associations in both datasets (one time including the Danish samples, the other time restricted to samples from Spain only), we can extract the association metrics from both <code>SIAMCAT</code> objects and compare them in a scatter plot.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">assoc.sp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/associations.html">associations</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span>
<span class="va">assoc.sp</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">assoc.sp</span><span class="op">)</span>
<span class="va">assoc.sp_dnk</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/associations.html">associations</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span>
<span class="va">assoc.sp_dnk</span><span class="op">$</span><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">assoc.sp_dnk</span><span class="op">)</span>

<span class="va">df.plot</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">assoc.sp</span>, <span class="va">assoc.sp_dnk</span>, by<span class="op">=</span><span class="st">'species'</span><span class="op">)</span>
<span class="va">df.plot</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>highlight<span class="op">=</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">species</span>, <span class="st">'formicigenerans'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">p.adj.x</span><span class="op">)</span>, y<span class="op">=</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">p.adj.y</span><span class="op">)</span>, col<span class="op">=</span><span class="va">highlight</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">xlab</span><span class="op">(</span><span class="st">'Spanish samples only\n-log10(q)'</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">ylab</span><span class="op">(</span><span class="st">'Spanish and Danish samples only\n-log10(q)'</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme</span><span class="op">(</span>panel.grid.major <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>colour<span class="op">=</span><span class="st">'lightgrey'</span><span class="op">)</span>,
          aspect.ratio <span class="op">=</span> <span class="fl">1.3</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">scale_colour_manual</span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'darkgrey'</span>, <span class="st">'#D41645'</span><span class="op">)</span>, guide<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">annotate</span><span class="op">(</span><span class="st">'text'</span>, x<span class="op">=</span><span class="fl">0.7</span>, y<span class="op">=</span><span class="fl">8</span>, label<span class="op">=</span><span class="st">'Dorea formicigenerans'</span><span class="op">)</span></code></pre></div>
<p><img src="SIAMCAT_confounder_files/figure-html/conf_assoc_plot-1.png" width="700"></p>
<p>This result shows that several species are only signficant if the Danish control samples are included, but not when considering only the Spanish samples.</p>
<p>As an example, we highlighted the species <em>“Dorea formicigenerans”</em> in the plot above. The test is not significant in the Spanish cohort, but is highly significant when the Danish samples are included.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># extract information out of the siamcat object</span>
<span class="va">feat.dnk</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get.filt_feat.matrix.html">get.filt_feat.matrix</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span>
<span class="va">label.dnk</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/label.html">label</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span><span class="op">$</span><span class="va">label</span>
<span class="va">country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/meta.html">meta</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span><span class="op">$</span><span class="va">Country</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">country</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="../reference/meta.html">meta</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span><span class="op">)</span>

<span class="va">df.plot</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>dorea<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">feat.dnk</span><span class="op">[</span><span class="fu">str_detect</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">feat.dnk</span><span class="op">)</span>,
                                                  <span class="st">'formicigenerans'</span><span class="op">)</span>,
                                       <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">label.dnk</span><span class="op">)</span><span class="op">]</span> <span class="op">+</span> <span class="fl">1e-05</span><span class="op">)</span>,
                  label<span class="op">=</span><span class="va">label.dnk</span>, country<span class="op">=</span><span class="va">country</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>label<span class="op">=</span><span class="fu">case_when</span><span class="op">(</span><span class="va">label</span><span class="op">==</span><span class="st">'-1'</span><span class="op">~</span><span class="st">'CTR'</span>, <span class="cn">TRUE</span><span class="op">~</span><span class="st">"UC"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>x_value<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">country</span>, <span class="st">'_'</span>, <span class="va">label</span><span class="op">)</span><span class="op">)</span>

<span class="va">df.plot</span> <span class="op">%&gt;%</span>
  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">x_value</span>, y<span class="op">=</span><span class="va">dorea</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">geom_boxplot</span><span class="op">(</span>outlier.shape <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">geom_jitter</span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.08</span>, stroke<span class="op">=</span><span class="fl">0</span>, alpha<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">xlab</span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">ylab</span><span class="op">(</span><span class="st">"log10(Dorea formicigenerans)"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/stat_compare_means.html">stat_compare_means</a></span><span class="op">(</span>comparisons <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'DNK_CTR'</span>, <span class="st">'ESP_CTR'</span><span class="op">)</span>,
                                          <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'DNK_CTR'</span>, <span class="st">'ESP_UC'</span><span class="op">)</span>,
                                          <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'ESP_CTR'</span>, <span class="st">'ESP_UC'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="SIAMCAT_confounder_files/figure-html/dorea_plot-1.png" width="700"></p>
</div>
<div id="machine-learning" class="section level2">
<h2 class="hasAnchor">
<a href="#machine-learning" class="anchor"></a>Machine Learning</h2>
<p>The results from the machine learning workflows can also be biased by the differences between countries, leading to exaggerated performance estimates.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, norm.method <span class="op">=</span> <span class="st">'log.std'</span>,
                                  norm.param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>log.n0<span class="op">=</span><span class="fl">1e-06</span>, sd.min.q<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span>
<span class="co">## Features normalized successfully.</span>
<span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, num.folds <span class="op">=</span> <span class="fl">5</span>, num.resample <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="co">## Features splitted for cross-validation successfully.</span>
<span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj.full</span>, method<span class="op">=</span><span class="st">'lasso'</span><span class="op">)</span>
<span class="co">## Trained lasso models successfully.</span>
<span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span>
<span class="co">## Made predictions successfully.</span>
<span class="va">sc.obj.full</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.full</span><span class="op">)</span>
<span class="co">## Evaluated predictions successfully.</span></code></pre></div>
<p>When we compare the performance of the two different models, the model with the Danish and Spanish samples included seems to perform better (higher AUROC value). However, the previous analysis suggests that this performance estimate is biased and exaggerated because differences between Spanish and Danish samples can be very large.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/model.evaluation.plot.html">model.evaluation.plot</a></span><span class="op">(</span><span class="st">"Spanish samples only"</span><span class="op">=</span><span class="va">sc.obj</span>,
                      <span class="st">"Danish and Spanish samples"</span><span class="op">=</span><span class="va">sc.obj.full</span>,
                      fn.plot <span class="op">=</span> <span class="st">'./eval_plot_dnk.pdf'</span><span class="op">)</span>
<span class="co">## Plotted evaluation of predictions successfully to: ./eval_plot_dnk.pdf</span></code></pre></div>
<p><img src="eval_plot_dnk.png"></p>
<p>To demonstrate how machine learning models can exploit this confounding factor, we can train a model to distinguish between Spanish and Danish control samples. As you can see, the model can distinguish between the two countries with almost perfect accuracy.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta.nielsen.country</span> <span class="op">&lt;-</span> <span class="va">meta.nielsen</span><span class="op">[</span><span class="va">meta.nielsen</span><span class="op">$</span><span class="va">Group</span><span class="op">==</span><span class="st">'CTR'</span>,<span class="op">]</span>

<span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat</span>, meta<span class="op">=</span><span class="va">meta.nielsen.country</span>,
                          label<span class="op">=</span><span class="st">'Country'</span>, case<span class="op">=</span><span class="st">'ESP'</span><span class="op">)</span>
<span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj.country</span>, cutoff<span class="op">=</span><span class="fl">1e-04</span>,
                              filter.method <span class="op">=</span> <span class="st">'abundance'</span><span class="op">)</span>
<span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj.country</span>, cutoff<span class="op">=</span><span class="fl">0.05</span>,
                              filter.method<span class="op">=</span><span class="st">'prevalence'</span>,
                              feature.type <span class="op">=</span> <span class="st">'filtered'</span><span class="op">)</span>
<span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">sc.obj.country</span>, norm.method <span class="op">=</span> <span class="st">'log.std'</span>,
                                     norm.param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>log.n0<span class="op">=</span><span class="fl">1e-06</span>,
                                                       sd.min.q<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span>
<span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span><span class="va">sc.obj.country</span>,
                                    num.folds <span class="op">=</span> <span class="fl">5</span>, num.resample <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj.country</span>, method<span class="op">=</span><span class="st">'lasso'</span><span class="op">)</span>
<span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj.country</span><span class="op">)</span>
<span class="va">sc.obj.country</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj.country</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="../reference/eval_data.html">eval_data</a></span><span class="op">(</span><span class="va">sc.obj.country</span><span class="op">)</span><span class="op">$</span><span class="va">auroc</span><span class="op">)</span>
<span class="co">## Area under the curve: 0.9701</span></code></pre></div>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## R version 4.0.2 (2020-06-22)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Catalina 10.15.5</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] ggpubr_0.4.0      SIAMCAT_1.11.1    phyloseq_1.32.0   mlr_2.18.0       </span>
<span class="co">##  [5] ParamHelpers_1.14 forcats_0.5.1     stringr_1.4.0     dplyr_1.0.4      </span>
<span class="co">##  [9] purrr_0.3.4       readr_1.4.0       tidyr_1.1.2       tibble_3.0.6     </span>
<span class="co">## [13] ggplot2_3.3.3     tidyverse_1.3.0   BiocStyle_2.16.1 </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] colorspace_2.0-0    ggsignif_0.6.0      rio_0.5.16         </span>
<span class="co">##   [4] ellipsis_0.3.1      rprojroot_2.0.2     XVector_0.28.0     </span>
<span class="co">##   [7] fs_1.5.0            rstudioapi_0.13     farver_2.0.3       </span>
<span class="co">##  [10] lubridate_1.7.9.2   xml2_1.3.2          PRROC_1.3.1        </span>
<span class="co">##  [13] codetools_0.2-18    splines_4.0.2       cachem_1.0.4       </span>
<span class="co">##  [16] knitr_1.31          ade4_1.7-16         jsonlite_1.7.2     </span>
<span class="co">##  [19] pROC_1.17.0.1       gridBase_0.4-7      broom_0.7.4        </span>
<span class="co">##  [22] cluster_2.1.1       dbplyr_2.1.0        BiocManager_1.30.10</span>
<span class="co">##  [25] compiler_4.0.2      httr_1.4.2          backports_1.2.1    </span>
<span class="co">##  [28] assertthat_0.2.1    Matrix_1.3-2        fastmap_1.1.0      </span>
<span class="co">##  [31] cli_2.3.0           htmltools_0.5.1.1   prettyunits_1.1.1  </span>
<span class="co">##  [34] tools_4.0.2         igraph_1.2.6        gtable_0.3.0       </span>
<span class="co">##  [37] glue_1.4.2          LiblineaR_2.10-8    reshape2_1.4.4     </span>
<span class="co">##  [40] fastmatch_1.1-0     Rcpp_1.0.6          parallelMap_1.5.0  </span>
<span class="co">##  [43] carData_3.0-4       Biobase_2.48.0      cellranger_1.1.0   </span>
<span class="co">##  [46] pkgdown_1.6.1.9000  vctrs_0.3.6         Biostrings_2.56.0  </span>
<span class="co">##  [49] multtest_2.44.0     ape_5.4-1           nlme_3.1-152       </span>
<span class="co">##  [52] iterators_1.0.13    xfun_0.21           openxlsx_4.2.3     </span>
<span class="co">##  [55] rvest_0.3.6         lifecycle_0.2.0     rstatix_0.7.0      </span>
<span class="co">##  [58] beanplot_1.2        zlibbioc_1.34.0     MASS_7.3-53.1      </span>
<span class="co">##  [61] scales_1.1.1        ragg_0.4.1          hms_1.0.0          </span>
<span class="co">##  [64] parallel_4.0.2      biomformat_1.16.0   rhdf5_2.32.4       </span>
<span class="co">##  [67] RColorBrewer_1.1-2  BBmisc_1.11         curl_4.3           </span>
<span class="co">##  [70] yaml_2.2.1          gridExtra_2.3       memoise_2.0.0      </span>
<span class="co">##  [73] stringi_1.5.3       highr_0.8           S4Vectors_0.26.1   </span>
<span class="co">##  [76] desc_1.2.0          corrplot_0.84       foreach_1.5.1      </span>
<span class="co">##  [79] checkmate_2.0.0     permute_0.9-5       BiocGenerics_0.34.0</span>
<span class="co">##  [82] zip_2.1.1           shape_1.4.5         matrixStats_0.58.0 </span>
<span class="co">##  [85] rlang_0.4.10        pkgconfig_2.0.3     systemfonts_1.0.1  </span>
<span class="co">##  [88] evaluate_0.14       lattice_0.20-41     Rhdf5lib_1.10.1    </span>
<span class="co">##  [91] labeling_0.4.2      tidyselect_1.1.0    plyr_1.8.6         </span>
<span class="co">##  [94] magrittr_2.0.1      bookdown_0.21       R6_2.5.0           </span>
<span class="co">##  [97] IRanges_2.22.2      generics_0.1.0      DBI_1.1.1          </span>
<span class="co">## [100] foreign_0.8-81      pillar_1.4.7        haven_2.3.1        </span>
<span class="co">## [103] withr_2.4.1         mgcv_1.8-33         abind_1.4-5        </span>
<span class="co">## [106] survival_3.2-7      car_3.0-10          modelr_0.1.8       </span>
<span class="co">## [109] crayon_1.4.1        rmarkdown_2.6       progress_1.2.2     </span>
<span class="co">## [112] grid_4.0.2          readxl_1.3.1        data.table_1.13.6  </span>
<span class="co">## [115] vegan_2.5-7         infotheo_1.2.0      reprex_1.0.0       </span>
<span class="co">## [118] digest_0.6.27       textshaping_0.3.0   glmnet_4.1         </span>
<span class="co">## [121] stats4_4.0.2        munsell_0.5.0</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Konrad Zych, Jakob Wirbel, Georg Zeller.</p>
</div>

<div class="privacy">
  <p><strong>Privacy:</strong> Usage of this site follows
      <a href="https://www.embl.de/aboutus/privacy_policy/index.html">EMBL’s Privacy Policy</a><br>
      In accordance with that policy, we use Matomo to collect anonymised data on visits to, downloads from, and searches of this site.
      Contact <a href="mailto:zeller@embl.de">Georg Zeller</a> for details.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
