<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SIAMCAT input files formats • SIAMCAT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SIAMCAT input files formats">
<meta property="og:description" content="SIAMCAT">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><script type="text/javascript">
  var _paq = _paq || [];
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//tr-denbi.embl.de/heimdall/";
    _paq.push(['setTrackerUrl', u+'p.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'p.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SIAMCAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SIAMCAT_vignette.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/SIAMCAT_vignette.html">Introduction to SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_confounder.html">SIAMCAT example with confounders</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_holdout.html">Holdout testing with SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_meta.html">SIAMCAT machine learning meta-analysis</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_ml_pitfalls.html">Machine learning pitfalls</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_read-in.html">SIAMCAT input formats</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zellerlab/siamcat" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>SIAMCAT input files formats</h1>
                        <h4 data-toc-skip class="author">Konrad Zych,
Jakob Wirbel, and Georg Zeller</h4>
            <address class="author_afil">
      EMBL
Heidelberg<br><a class="author_email" href="mailto:#"></a><a href="mailto:georg.zeller@embl.de" class="email">georg.zeller@embl.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">Date last modified:
2018-09-24</h4>
      
      
      <div class="hidden name"><code>SIAMCAT_read-in.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette illustrates how to read and input your own data to the
<code>SIAMCAT</code> package. We will cover reading in text files from
the disk, formatting them and using them to create an object of
<code>siamcat-class</code>.</p>
<p>The <code>siamcat-class</code> is the centerpiece of the package. All
of the input data and result are stored inside of it. The structure of
the object is described below in the <a href="#siamcat-class-object">siamcat-class object</a> section.</p>
</div>
<div class="section level2">
<h2 id="loading-your-data-into-r">Loading your data into R<a class="anchor" aria-label="anchor" href="#loading-your-data-into-r"></a>
</h2>
<div class="section level3">
<h3 id="siamcat-input">SIAMCAT input<a class="anchor" aria-label="anchor" href="#siamcat-input"></a>
</h3>
<p>Generally, there are three types of input for
<code>SIAMCAT</code>:</p>
<div class="section level4">
<h4 id="features">Features<a class="anchor" aria-label="anchor" href="#features"></a>
</h4>
<p>The features should be a <code>matrix</code>, a
<code>data.frame</code>, or an <code>otu_table</code>, organized as
follows:</p>
<p><code>features (in rows) x samples (in columns).</code></p>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">Sample_1</th>
<th align="right">Sample_2</th>
<th align="right">Sample_3</th>
<th align="right">Sample_4</th>
<th align="right">Sample_5</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Feature_1</strong></td>
<td align="right">0.59</td>
<td align="right">0.71</td>
<td align="right">0.78</td>
<td align="right">0.61</td>
<td align="right">0.66</td>
</tr>
<tr class="even">
<td><strong>Feature_2</strong></td>
<td align="right">0.00</td>
<td align="right">0.02</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td><strong>Feature_3</strong></td>
<td align="right">0.02</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.20</td>
</tr>
<tr class="even">
<td><strong>Feature_4</strong></td>
<td align="right">0.34</td>
<td align="right">0.00</td>
<td align="right">0.13</td>
<td align="right">0.07</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td><strong>Feature_5</strong></td>
<td align="right">0.06</td>
<td align="right">0.16</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
</tbody>
</table>
<blockquote>
<p>Please note that <code>SIAMCAT</code> is supposed to work with
<strong>relative abundances</strong>. Other types of data (e.g. counts)
will also work, but not all functions of the package will result in
meaningful outputs.</p>
</blockquote>
<p>An example of a typical feature file is attached to the
<code>SIAMCAT</code> package, containing data from a publication
investigating the microbiome in colorectal cancer (CRC) patients and
controls (the study can be found here: <a href="http://europepmc.org/abstract/MED/25432777" class="external-link">Zeller et al</a>). The
metagenomics data were processed with the <a href="http://mocat.embl.de/" class="external-link">MOCAT</a> pipeline, returning taxonomic
profiles on the species levels (<code>specI</code>):</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SIAMCAT</span><span class="op">)</span></span>
<span><span class="va">fn.in.feat</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>,</span>
<span>    <span class="st">"feat_crc_zeller_msb_mocat_specI.tsv"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"SIAMCAT"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>One way to load such data into <code>R</code> could be the use of
<code>read.table</code></p>
<p><em>(Beware of the defaults in R! They are not always
useful…)</em></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">fn.in.feat</span>, sep<span class="op">=</span><span class="st">'\t'</span>,</span>
<span>    header<span class="op">=</span><span class="cn">TRUE</span>, quote<span class="op">=</span><span class="st">''</span>,</span>
<span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># look at some features</span></span>
<span><span class="va">feat</span><span class="op">[</span><span class="fl">110</span><span class="op">:</span><span class="fl">114</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##                                CCIS27304052ST-3-0 CCIS15794887ST-4-0</span></span>
<span><span class="co">## Bacteroides caccae [h:1096]          1.557937e-03       1.761949e-03</span></span>
<span><span class="co">## Bacteroides eggerthii [h:1097]       2.734527e-05       4.146882e-05</span></span>
<span><span class="co">## Bacteroides stercoris [h:1098]       1.173786e-03       2.475838e-03</span></span>
<span><span class="co">## Bacteroides clarus [h:1099]          4.830533e-04       4.589747e-06</span></span>
<span><span class="co">## Methanohalophilus mahii [h:11]       0.000000e+00       0.000000e+00</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="metadata">Metadata<a class="anchor" aria-label="anchor" href="#metadata"></a>
</h4>
<p>The metadata should be either a matrix or a
<code>data.frame</code>.</p>
<p><code>samples (in rows) x metadata (in columns)</code>:</p>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">Age</th>
<th align="right">Gender</th>
<th align="right">BMI</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Sample_1</strong></td>
<td align="right">52</td>
<td align="right">1</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td><strong>Sample_2</strong></td>
<td align="right">37</td>
<td align="right">1</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td><strong>Sample_3</strong></td>
<td align="right">66</td>
<td align="right">2</td>
<td align="right">24</td>
</tr>
<tr class="even">
<td><strong>Sample_4</strong></td>
<td align="right">54</td>
<td align="right">2</td>
<td align="right">26</td>
</tr>
<tr class="odd">
<td><strong>Sample_5</strong></td>
<td align="right">65</td>
<td align="right">2</td>
<td align="right">30</td>
</tr>
</tbody>
</table>
<p>The <code>rownames</code> of the metadata should match the
<code>colnames</code> of the feature matrix.</p>
<p>Again, an example of such a file is attached to the
<code>SIAMCAT</code> package, taken from the same study:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fn.in.meta</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>,</span>
<span>    <span class="st">"num_metadata_crc_zeller_msb_mocat_specI.tsv"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"SIAMCAT"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Also here, the <code>read.table</code> can be used to load the data
into <code>R</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">fn.in.meta</span>, sep<span class="op">=</span><span class="st">'\t'</span>,</span>
<span>    header<span class="op">=</span><span class="cn">TRUE</span>, quote<span class="op">=</span><span class="st">''</span>,</span>
<span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    age gender bmi diagnosis localization crc_stage fobt</span></span>
<span><span class="co">## CCIS27304052ST-3-0  52      1  20         0           NA         0    0</span></span>
<span><span class="co">## CCIS15794887ST-4-0  37      1  18         0           NA         0    0</span></span>
<span><span class="co">## CCIS74726977ST-3-0  66      2  24         1           NA         0    0</span></span>
<span><span class="co">## CCIS16561622ST-4-0  54      2  26         0           NA         0    0</span></span>
<span><span class="co">## CCIS79210440ST-3-0  65      2  30         0           NA         0    1</span></span>
<span><span class="co">## CCIS82507866ST-3-0  57      2  24         0           NA         0    0</span></span>
<span><span class="co">##                    wif_test</span></span>
<span><span class="co">## CCIS27304052ST-3-0        0</span></span>
<span><span class="co">## CCIS15794887ST-4-0        0</span></span>
<span><span class="co">## CCIS74726977ST-3-0       NA</span></span>
<span><span class="co">## CCIS16561622ST-4-0        0</span></span>
<span><span class="co">## CCIS79210440ST-3-0        0</span></span>
<span><span class="co">## CCIS82507866ST-3-0        0</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="label">Label<a class="anchor" aria-label="anchor" href="#label"></a>
</h4>
<p>Finally, the label can come in different three different
flavours:</p>
<ul>
<li>
<p><strong>Named vector</strong>: A named vector containing
information about cases and controls. The names of the vector should
match the <code>rownames</code> of the metadata and the
<code>colnames</code> of the feature data. The label can contain either
the information about cases and controls either</p>
<ul>
<li>as integers (e.g. <code>0</code> and <code>1</code>),</li>
<li>as characters (e.g. <code>CTR</code> and <code>IBD</code>), or</li>
<li>as factors.</li>
</ul>
</li>
<li><p><strong>Metadata column</strong>: You can provide the name of a
column in the metadata for the creation of the label. See below for an
example.</p></li>
<li>
<p><strong>Label file</strong>: <code>SIAMCAT</code> has a function
called <code>read.label</code>, which will create a label object from a
label file. The file should be organized as follows:</p>
<ul>
<li>The first line is supposed to read:
<code>#BINARY:1=[label for cases];-1=[label for controls]</code>
</li>
<li>The second row should contain the sample identifiers as
tab-separated list (consistent with feature and metadata).</li>
<li>The third row is then supposed to contain the actual class labels
(tab-separated): <code>1</code> for each case and <code>-1</code> for
each control.</li>
</ul>
<p>An example file is attached to the package again, if you want to have
a look at it.</p>
</li>
</ul>
<p>For our example dataset, we can create the label object out of the
metadata column called <code>diagnosis</code>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">label</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.label.html">create.label</a></span><span class="op">(</span>meta<span class="op">=</span><span class="va">meta</span>, label<span class="op">=</span><span class="st">"diagnosis"</span>,</span>
<span>    case <span class="op">=</span> <span class="fl">1</span>, control<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p>When we later plot the results, it might be nicer to have names for
the different groups stored in the label object (instead of
<code>1</code> and <code>0</code>). We can also supply them to the
<code>create.label</code> function:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">label</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.label.html">create.label</a></span><span class="op">(</span>meta<span class="op">=</span><span class="va">meta</span>, label<span class="op">=</span><span class="st">"diagnosis"</span>,</span>
<span>    case <span class="op">=</span> <span class="fl">1</span>, control<span class="op">=</span><span class="fl">0</span>,</span>
<span>    p.lab <span class="op">=</span> <span class="st">'cancer'</span>, n.lab <span class="op">=</span> <span class="st">'healthy'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Label used as case:</span></span>
<span><span class="co">##    1</span></span>
<span><span class="co">## Label used as control:</span></span>
<span><span class="co">##    0</span></span></code></pre>
<pre><code><span><span class="co">## + finished create.label.from.metadata in 0.001 s</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">label</span><span class="op">$</span><span class="va">info</span></span></code></pre></div>
<pre><code><span><span class="co">## healthy  cancer </span></span>
<span><span class="co">##      -1       1</span></span></code></pre>
<blockquote>
<p>Note:<br>
If you have no label information for your dataset, you can still create
a <code>SIAMCAT</code> object from your features alone. The
<code>SIAMCAT</code> object without label information will contain a
<code>TEST</code> label that can be used for making holdout predictions.
Other functions, e.g. model training, will not work on such an
object.</p>
</blockquote>
</div>
</div>
<div class="section level3">
<h3 id="lefse-format-files">LEfSe format files<a class="anchor" aria-label="anchor" href="#lefse-format-files"></a>
</h3>
<p><a href="https://bitbucket.org/biobakery/biobakery/wiki/lefse" class="external-link">LEfSe</a> is
a tool for identification of associations between micriobial features
and up to two metadata. LEfSe uses LDA (linear discriminant
analysis).</p>
<p>LEfSe input file is a <code>.tsv</code> file. The first few rows
contain the metadata. The following row contains sample names and the
rest of the rows are occupied by features. The first column holds the
row names:</p>
<table class="table">
<colgroup>
<col width="13%">
<col width="17%">
<col width="17%">
<col width="17%">
<col width="17%">
<col width="17%">
</colgroup>
<thead><tr class="header">
<th>label</th>
<th align="right">healthy</th>
<th align="right">healthy</th>
<th align="right">healthy</th>
<th align="right">cancer</th>
<th align="right">cancer</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>age</strong></td>
<td align="right">52</td>
<td align="right">37</td>
<td align="right">66</td>
<td align="right">54</td>
<td align="right">65</td>
</tr>
<tr class="even">
<td><strong>gender</strong></td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td><strong>Sample_info</strong></td>
<td align="right">Sample_1</td>
<td align="right">Sample_2</td>
<td align="right">Sample_3</td>
<td align="right">Sample_4</td>
<td align="right">Sample_5</td>
</tr>
<tr class="even">
<td><strong>Feature_1</strong></td>
<td align="right">0.59</td>
<td align="right">0.71</td>
<td align="right">0.78</td>
<td align="right">0.61</td>
<td align="right">0.66</td>
</tr>
<tr class="odd">
<td><strong>Feature_2</strong></td>
<td align="right">0.00</td>
<td align="right">0.02</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="even">
<td><strong>Feature_3</strong></td>
<td align="right">0.02</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td><strong>Feature_4</strong></td>
<td align="right">0.34</td>
<td align="right">0.00</td>
<td align="right">0.43</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="even">
<td><strong>Feature_5</strong></td>
<td align="right">0.56</td>
<td align="right">0.56</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
</tbody>
</table>
<p>An example of such a file is attached to the <code>SIAMCAT</code>
package:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fn.in.lefse</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>,</span>
<span>    <span class="st">"LEfSe_crc_zeller_msb_mocat_specI.tsv"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"SIAMCAT"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><code>SIAMCAT</code> has a dedicated function to read LEfSe format
files. The <code>read.lefse</code> function will read in the input file
and extract metadata and features:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.and.features</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read.lefse.html">read.lefse</a></span><span class="op">(</span><span class="va">fn.in.lefse</span>,</span>
<span>    rows.meta <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, row.samples <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">meta.and.features</span><span class="op">$</span><span class="va">meta</span></span>
<span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="va">meta.and.features</span><span class="op">$</span><span class="va">feat</span></span></code></pre></div>
<p>We can then create a label object from one of the columns of the meta
object and create a <code>siamcat</code> object:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">label</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.label.html">create.label</a></span><span class="op">(</span>meta<span class="op">=</span><span class="va">meta</span>, label<span class="op">=</span><span class="st">"label"</span>, case <span class="op">=</span> <span class="st">"cancer"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Label used as case:</span></span>
<span><span class="co">##    cancer</span></span>
<span><span class="co">## Label used as control:</span></span>
<span><span class="co">##    healthy</span></span></code></pre>
<pre><code><span><span class="co">## + finished create.label.from.metadata in 0.049 s</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="metagenomeseq-format-files">metagenomeSeq format files<a class="anchor" aria-label="anchor" href="#metagenomeseq-format-files"></a>
</h3>
<p><a href="http://bioconductor.org/packages/metagenomeSeq/" class="external-link">metagenomeSeq</a>
is an R package to determine differentially abundant features between
multiple samples.</p>
<p>There are two ways to input data into metagenomeSeq:</p>
<ol style="list-style-type: lower-alpha">
<li>two files, one for metadata and one for features - those can be used
in <code>SIAMCAT</code> just like described in <a href="#SIAMCAT-input">SIAMCAT input</a> with
<code>read.table</code>:</li>
</ol>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fn.in.feat</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>,</span>
<span>    <span class="st">"CHK_NAME.otus.count.csv"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"metagenomeSeq"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">feat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">fn.in.feat</span>, sep<span class="op">=</span><span class="st">'\t'</span>,</span>
<span>    header<span class="op">=</span><span class="cn">TRUE</span>, quote<span class="op">=</span><span class="st">''</span>, row.names <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li>
<code>BIOM</code> format file, that can be used in
<code>SIAMCAT</code> as described in the <a href="BIOM-format-files">following section</a>
</li>
</ol>
</div>
<div class="section level3">
<h3 id="biom-format-files">BIOM format files<a class="anchor" aria-label="anchor" href="#biom-format-files"></a>
</h3>
<p>The BIOM format files can be added to <code>SIAMCAT</code> via
<code>phyloseq</code>. First the file should be imported using the
<code>phyloseq</code> function <code>import_biom</code>. Then a
<code>phyloseq</code> object can be imported as a <code>siamcat</code>
object as descibed in the <a href="#Creating-a-siamcat-object-of-a-phyloseq-object">next
section.</a></p>
</div>
<div class="section level3">
<h3 id="creating-a-siamcat-object-of-a-phyloseq-object">Creating a siamcat object of a phyloseq object<a class="anchor" aria-label="anchor" href="#creating-a-siamcat-object-of-a-phyloseq-object"></a>
</h3>
<p>The <code>siamcat</code> object extends on the <code>phyloseq</code>
object. Therefore, creating a <code>siamcat</code> object from a
<code>phyloseq</code> object is really straightforward. This can be done
with the <code>siamcat</code> constructor function. First, however, we
need to create a label object:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"GlobalPatterns"</span><span class="op">)</span> <span class="co">## phyloseq example data</span></span>
<span><span class="va">label</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.label.html">create.label</a></span><span class="op">(</span>meta<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span>,</span>
<span>    label <span class="op">=</span> <span class="st">"SampleType"</span>,</span>
<span>    case <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Freshwater"</span>, <span class="st">"Freshwater (creek)"</span>, <span class="st">"Ocean"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Label used as case:</span></span>
<span><span class="co">##    Freshwater,Freshwater (creek),Ocean</span></span>
<span><span class="co">## Label used as control:</span></span>
<span><span class="co">##    rest</span></span></code></pre>
<pre><code><span><span class="co">## + finished create.label.from.metadata in 0.003 s</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run the constructor function</span></span>
<span><span class="va">siamcat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>phyloseq<span class="op">=</span><span class="va">GlobalPatterns</span>, label<span class="op">=</span><span class="va">label</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## + starting validate.data</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking overlap between labels and features</span></span></code></pre>
<pre><code><span><span class="co">## + Keeping labels of 26 sample(s).</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking sample number per class</span></span></code></pre>
<pre><code><span><span class="co">## Data set has a limited number of training examples:</span></span>
<span><span class="co">##  rest    18 </span></span>
<span><span class="co">##  Case    8 </span></span>
<span><span class="co">## Note that a dataset this small/skewed is not necessarily suitable for analysis in this pipeline.</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking overlap between samples and metadata</span></span></code></pre>
<pre><code><span><span class="co">## + finished validate.data in 0.264 s</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="creating-a-siamcat-class-object">Creating a siamcat-class object<a class="anchor" aria-label="anchor" href="#creating-a-siamcat-class-object"></a>
</h2>
<p>The <code>siamcat-class</code> is the centerpiece of the package. All
of the is stored inside of the object: <img src="siamcat.png" alt="internal make-up of a siamcat object"></p>
<p>In the figure above, rectangles depict slots of the object and the
class of the object stored in the slot is given in the ovals. There are
two obligatory slots -<strong>phyloseq</strong> (containing the metadata
as <code>sample_data</code> and the original features as
<code>otu_table</code>) and <strong>label</strong> - marked with thick
borders.</p>
<p>The <code>siamcat</code> object is constructed using the
<code><a href="../reference/siamcat.html">siamcat()</a></code> function. There are two ways to initialize
it:</p>
<ul>
<li>
<p><strong>Features</strong>: You can provide a feature
<code>matrix</code>, <code>data.frame</code>, or <code>otu_table</code>
to the function (together with label and metadata information):</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat</span>, label<span class="op">=</span><span class="va">label</span>, meta<span class="op">=</span><span class="va">meta</span><span class="op">)</span></span></code></pre></div>
</li>
<li>
<p><strong>phyloseq</strong>: The alternative is to create a
<code>siamcat</code> object directly out of a <code>phyloseq</code>
object:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">siamcat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>phyloseq<span class="op">=</span><span class="va">phyloseq</span>, label<span class="op">=</span><span class="va">label</span><span class="op">)</span></span></code></pre></div>
</li>
</ul>
<p>Please note that you <strong>have to</strong> provide either
<code>feat</code> or <code>phyloseq</code> and that you
<strong>cannot</strong> provide both.</p>
<p>In order to explain the <code>siamcat</code> object better we will
show how each of the slots is filled.</p>
<div class="section level3">
<h3 id="phyloseq-label-and-orig_feat-slots">phyloseq, label and orig_feat slots<a class="anchor" aria-label="anchor" href="#phyloseq-label-and-orig_feat-slots"></a>
</h3>
<p>The phyloseq and label slots are obligatory.</p>
<ul>
<li>The phyloseq slot is an object of class <code>phyloseq</code>, which
is described in the help file of the <code>phyloseq</code> class. Help
can be accessed by typing into R console:
<code><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq-class.html" class="external-link">help('phyloseq-class')</a></code>.
<ul>
<li>The <code>otu_table</code> slot in <code>phyloseq</code> -see
<code><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-class.html" class="external-link">help('otu_table-class')</a></code>- stores the original feature table.
For <code>SIAMCAT</code>, this slot can be accessed by
<code>orig_feat</code>.</li>
</ul>
</li>
<li>The label slot contains a list. This list has a specific set of
entries -see <code>help('label-class')</code>- that are automatically
generated by the <code>read.label</code> or <code>create.label</code>
functions.</li>
</ul>
<p>The <code>phyloseq</code>, label and orig_feat are filled when the
<code>siamcat</code> object is first created with the constructor
function. <img src="allCr.png" alt="construction"></p>
</div>
<div class="section level3">
<h3 id="all-the-other-slots">All the other slots<a class="anchor" aria-label="anchor" href="#all-the-other-slots"></a>
</h3>
<p>Other slots are filled during the run of the <code>SIAMCAT</code>
workflow: <img src="Slots_create.png" alt="workflow"></p>
</div>
<div class="section level3">
<h3 id="accessing-and-assigning-slots">Accessing and assigning slots<a class="anchor" aria-label="anchor" href="#accessing-and-assigning-slots"></a>
</h3>
<p>Each slot in <code>siamcat</code> can be accessed by typing</p>
<pre><code><span><span class="fu">slot_name</span><span class="op">(</span><span class="va">siamcat</span><span class="op">)</span></span></code></pre>
<p>e.g. for the <code>eval_data</code> slot you can types</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/eval_data-methods.html">eval_data</a></span><span class="op">(</span><span class="va">siamcat</span><span class="op">)</span></span></code></pre></div>
<p>There is one notable exception: the phyloseq slot has to be accessed
with <code>physeq(siamcat)</code> due to technical reasons.</p>
<p>Slots will be filled during the <code>SIAMCAT</code> workflow by the
package’s functions. However, if for any reason a slot needs to be
assigned outside of the workflow, the following formula can be used:</p>
<pre><code><span><span class="fu">slot_name</span><span class="op">(</span><span class="va">siamcat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">object_to_assign</span></span></code></pre>
<p>e.g. to assign a <code>new_label</code> object to the
<code>label</code> slot:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/label-methods.html">label</a></span><span class="op">(</span><span class="va">siamcat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">new_label</span></span></code></pre></div>
<p><em>Please note that this may lead to unforeseen
consequences…</em></p>
</div>
<div class="section level3">
<h3 id="slots-inside-the-slots">Slots inside the slots<a class="anchor" aria-label="anchor" href="#slots-inside-the-slots"></a>
</h3>
<p>There are two slots that have slots inside of them. First, the
<code>model_list</code> slot has a <code>models</code> slot that
contains the actual list of <a href="https://mlr-org.github.io/mlr-tutorial/devel/html/index.html" class="external-link">mlr</a>
models -can be accessed via <code>models(siamcat)</code>- and
<code>model.type</code> which is a character with the name of the method
used to train the model: <code>model_type(siamcat)</code>.</p>
<p>The phyloseq slot has a complex structure. However, unless the
phyloseq object is created outside of the <code>SIAMCAT</code> workflow,
only two slots of phyloseq slot will be occupied: the
<code>otu_table</code> slot containing the features table and the
<code>sam_data</code> slot containing metadata information. Both can be
accessed by typing either <code>features(siamcat)</code> or
<code>meta(siamcat)</code>.</p>
<p>Additional slots inside the phyloseq slots do not have dedicated
accessors, but can easily be reached once the phyloseq object is
exported from the <code>siamcat</code> object:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">phyloseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/physeq-methods.html">physeq</a></span><span class="op">(</span><span class="va">siamcat</span><span class="op">)</span></span>
<span><span class="va">tax_tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">phyloseq</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tax_tab</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Taxonomy Table:     [6 taxa by 7 taxonomic ranks]:</span></span>
<span><span class="co">##        Kingdom   Phylum          Class          Order          Family         </span></span>
<span><span class="co">## 549322 "Archaea" "Crenarchaeota" "Thermoprotei" NA             NA             </span></span>
<span><span class="co">## 522457 "Archaea" "Crenarchaeota" "Thermoprotei" NA             NA             </span></span>
<span><span class="co">## 951    "Archaea" "Crenarchaeota" "Thermoprotei" "Sulfolobales" "Sulfolobaceae"</span></span>
<span><span class="co">## 244423 "Archaea" "Crenarchaeota" "Sd-NA"        NA             NA             </span></span>
<span><span class="co">## 586076 "Archaea" "Crenarchaeota" "Sd-NA"        NA             NA             </span></span>
<span><span class="co">## 246140 "Archaea" "Crenarchaeota" "Sd-NA"        NA             NA             </span></span>
<span><span class="co">##        Genus        Species                   </span></span>
<span><span class="co">## 549322 NA           NA                        </span></span>
<span><span class="co">## 522457 NA           NA                        </span></span>
<span><span class="co">## 951    "Sulfolobus" "Sulfolobusacidocaldarius"</span></span>
<span><span class="co">## 244423 NA           NA                        </span></span>
<span><span class="co">## 586076 NA           NA                        </span></span>
<span><span class="co">## 246140 NA           NA</span></span></code></pre>
<p>If you want to find out more about the phyloseq data structure, head
over to the <a href="https://bioconductor.org/packages/release/bioc/html/phyloseq.html" class="external-link">phyloseq</a>
BioConductor page. # Session Info</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] SIAMCAT_2.3.3    phyloseq_1.42.0  mlr3_0.14.1      BiocStyle_2.26.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] paradox_0.11.0         minqa_1.2.5            colorspace_2.1-0      </span></span>
<span><span class="co">##   [4] ellipsis_0.3.2         rprojroot_2.0.3        XVector_0.38.0        </span></span>
<span><span class="co">##   [7] fs_1.6.1               rstudioapi_0.14        listenv_0.9.0         </span></span>
<span><span class="co">##  [10] mlr3tuning_0.18.0      fansi_1.0.4            codetools_0.2-19      </span></span>
<span><span class="co">##  [13] splines_4.2.2          mlr3learners_0.5.6     PRROC_1.3.1           </span></span>
<span><span class="co">##  [16] cachem_1.0.7           knitr_1.42             ade4_1.7-22           </span></span>
<span><span class="co">##  [19] jsonlite_1.8.4         nloptr_2.0.3           pROC_1.18.0           </span></span>
<span><span class="co">##  [22] gridBase_0.4-7         cluster_2.1.4          BiocManager_1.30.20   </span></span>
<span><span class="co">##  [25] compiler_4.2.2         backports_1.4.1        Matrix_1.5-3          </span></span>
<span><span class="co">##  [28] fastmap_1.1.1          cli_3.6.0              prettyunits_1.1.1     </span></span>
<span><span class="co">##  [31] htmltools_0.5.4        tools_4.2.2            lmerTest_3.1-3        </span></span>
<span><span class="co">##  [34] igraph_1.4.1           gtable_0.3.1           glue_1.6.2            </span></span>
<span><span class="co">##  [37] GenomeInfoDbData_1.2.9 reshape2_1.4.4         LiblineaR_2.10-22     </span></span>
<span><span class="co">##  [40] dplyr_1.1.0            Rcpp_1.0.10            Biobase_2.58.0        </span></span>
<span><span class="co">##  [43] jquerylib_0.1.4        pkgdown_2.0.7          vctrs_0.5.2           </span></span>
<span><span class="co">##  [46] Biostrings_2.66.0      rhdf5filters_1.10.0    multtest_2.54.0       </span></span>
<span><span class="co">##  [49] ape_5.7-1              nlme_3.1-162           iterators_1.0.14      </span></span>
<span><span class="co">##  [52] xfun_0.37              stringr_1.5.0          globals_0.16.2        </span></span>
<span><span class="co">##  [55] lme4_1.1-32            lifecycle_1.0.3        beanplot_1.3.1        </span></span>
<span><span class="co">##  [58] future_1.32.0          zlibbioc_1.44.0        MASS_7.3-58.3         </span></span>
<span><span class="co">##  [61] scales_1.2.1           lgr_0.4.4              hms_1.1.2             </span></span>
<span><span class="co">##  [64] ragg_1.2.5             parallel_4.2.2         biomformat_1.26.0     </span></span>
<span><span class="co">##  [67] rhdf5_2.42.0           RColorBrewer_1.1-3     yaml_2.3.7            </span></span>
<span><span class="co">##  [70] memoise_2.0.1          gridExtra_2.3          ggplot2_3.4.1         </span></span>
<span><span class="co">##  [73] sass_0.4.5             stringi_1.7.12         S4Vectors_0.36.2      </span></span>
<span><span class="co">##  [76] desc_1.4.2             corrplot_0.92          foreach_1.5.2         </span></span>
<span><span class="co">##  [79] checkmate_2.1.0        permute_0.9-7          palmerpenguins_0.1.1  </span></span>
<span><span class="co">##  [82] BiocGenerics_0.44.0    boot_1.3-28.1          shape_1.4.6           </span></span>
<span><span class="co">##  [85] GenomeInfoDb_1.34.9    matrixStats_0.63.0     rlang_1.1.0           </span></span>
<span><span class="co">##  [88] pkgconfig_2.0.3        systemfonts_1.0.4      bitops_1.0-7          </span></span>
<span><span class="co">##  [91] evaluate_0.20          lattice_0.20-45        purrr_1.0.1           </span></span>
<span><span class="co">##  [94] Rhdf5lib_1.20.0        tidyselect_1.2.0       parallelly_1.34.0     </span></span>
<span><span class="co">##  [97] plyr_1.8.8             magrittr_2.0.3         bookdown_0.33         </span></span>
<span><span class="co">## [100] R6_2.5.1               IRanges_2.32.0         generics_0.1.3        </span></span>
<span><span class="co">## [103] DBI_1.1.3              pillar_1.8.1           mgcv_1.8-42           </span></span>
<span><span class="co">## [106] survival_3.5-5         RCurl_1.98-1.10        tibble_3.2.0          </span></span>
<span><span class="co">## [109] crayon_1.5.2           uuid_1.1-0             utf8_1.2.3            </span></span>
<span><span class="co">## [112] rmarkdown_2.20         progress_1.2.2         grid_4.2.2            </span></span>
<span><span class="co">## [115] data.table_1.14.8      vegan_2.6-4            infotheo_1.2.0.1      </span></span>
<span><span class="co">## [118] mlr3misc_0.11.0        bbotk_0.7.2            digest_0.6.31         </span></span>
<span><span class="co">## [121] numDeriv_2016.8-1.1    textshaping_0.3.6      stats4_4.2.2          </span></span>
<span><span class="co">## [124] munsell_0.5.0          glmnet_4.1-6           bslib_0.4.2</span></span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Konrad Zych, Jakob Wirbel, Georg Zeller.</p>
</div>

<div class="privacy">
  <p><strong>Privacy:</strong> Usage of this site follows
      <a href="https://www.embl.de/aboutus/privacy_policy/index.html" class="external-link">EMBL’s Privacy Policy</a><br>
      In accordance with that policy, we use Matomo to collect anonymised data on visits to, downloads from, and searches of this site.
      Contact <a href="mailto:zeller@embl.de">Georg Zeller</a> for details.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
