<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SIAMCAT: Statistical Inference of Associations between Microbial Communities And host phenoTypes • SIAMCAT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SIAMCAT: Statistical Inference of Associations between Microbial Communities And host phenoTypes">
<meta property="og:description" content="SIAMCAT">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><script type="text/javascript">
  var _paq = _paq || [];
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//tr-denbi.embl.de/heimdall/";
    _paq.push(['setTrackerUrl', u+'p.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'p.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SIAMCAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SIAMCAT_vignette.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/SIAMCAT_vignette.html">Introduction to SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_confounder.html">SIAMCAT example with confounders</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_holdout.html">Holdout testing with SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_meta.html">SIAMCAT machine learning meta-analysis</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_ml_pitfalls.html">Machine learning pitfalls</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_read-in.html">SIAMCAT input formats</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zellerlab/siamcat" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>SIAMCAT: Statistical Inference of Associations
between Microbial Communities And host phenoTypes</h1>
                        <h4 data-toc-skip class="author">Konrad Zych,
Jakob Wirbel, and Georg Zeller</h4>
            <address class="author_afil">
      EMBL
Heidelberg<br><a class="author_email" href="mailto:#"></a><a href="mailto:georg.zeller@embl.de" class="email">georg.zeller@embl.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">Date last modified:
2020-04-04</h4>
      
      
      <div class="hidden name"><code>SIAMCAT_vignette.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="about-this-vignette">About This Vignette<a class="anchor" aria-label="anchor" href="#about-this-vignette"></a>
</h2>
<p>This vignette aims to be a short tutorial for the main
functionalities of <code>SIAMCAT</code>. Examples of additional
workflows or more detailed tutorials can be found in other vignettes
(see the <a href="https://bioconductor.org/packages/SIAMCAT" class="external-link">BioConductor
page</a>).</p>
<p><code>SIAMCAT</code> is part of the suite of computational microbiome
analysis tools hosted at <a href="https://www.embl.org" class="external-link">EMBL</a> by the
groups of <a href="https://www.embl.de/research/units/scb/bork/index.html" class="external-link">Peer
Bork</a> and <a href="https://www.embl.de/research/units/scb/zeller/index.html" class="external-link">Georg
Zeller</a>. Find out more at <a href="http://microbiome-tools.embl.de/" class="external-link">EMBL-microbiome tools</a>.</p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Associations between microbiome and host phenotypes are ideally
described by quantitative models able to predict host status from
microbiome composition. <code>SIAMCAT</code> can do so for data from
hundreds of thousands of microbial taxa, gene families, or metabolic
pathways over hundreds of samples. <code>SIAMCAT</code> produces
graphical output for convenient assessment of the quality of the input
data and statistical associations, for model diagnostics and inference
revealing the most predictive microbial biomarkers.</p>
</div>
<div class="section level2">
<h2 id="quick-start">Quick Start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<p>For this vignette, we use an example dataset included in the
<code>SIAMCAT</code> package. As example dataset we use the data from
the publication of <a href="http://europepmc.org/abstract/MED/25432777" class="external-link">Zeller et al</a>,
which demonstrated the potential of microbial species in fecal samples
to distinguish patients with colorectal cancer (CRC) from healthy
controls.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SIAMCAT"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"feat_crc_zeller"</span>, package<span class="op">=</span><span class="st">"SIAMCAT"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"meta_crc_zeller"</span>, package<span class="op">=</span><span class="st">"SIAMCAT"</span><span class="op">)</span></span></code></pre></div>
<p>First, <code>SIAMCAT</code> needs a feature matrix (can be either a
<code>matrix</code>, a <code>data.frame</code>, or a
<code>phyloseq-otu_table</code>), which contains values of different
features (in rows) for different samples (in columns). For example, the
feature matrix included here contains relative abundances for bacterial
species calculated with the <a href="motu-tool.org">mOTU profiler</a>
for 141 samples:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">feat.crc.zeller</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##                                  CCIS27304052ST-3-0 CCIS15794887ST-4-0</span></span>
<span><span class="co">## UNMAPPED                                   0.589839          0.7142157</span></span>
<span><span class="co">## Methanoculleus marisnigri [h:1]            0.000000          0.0000000</span></span>
<span><span class="co">## Methanococcoides burtonii [h:10]           0.000000          0.0000000</span></span>
<span><span class="co">##                                  CCIS74726977ST-3-0</span></span>
<span><span class="co">## UNMAPPED                                  0.7818674</span></span>
<span><span class="co">## Methanoculleus marisnigri [h:1]           0.0000000</span></span>
<span><span class="co">## Methanococcoides burtonii [h:10]          0.0000000</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">feat.crc.zeller</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1754  141</span></span></code></pre>
<blockquote>
<p>Please note that <code>SIAMCAT</code> is supposed to work with
<strong>relative abundances</strong>. Other types of data (e.g. counts)
will also work, but not all functions of the package will result in
meaningful outputs.</p>
</blockquote>
<p>Secondly, we also have metadata about the samples in another
<code>data.frame</code>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">meta.crc.zeller</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    Age BMI Gender AJCC_stage     FOBT Group</span></span>
<span><span class="co">## CCIS27304052ST-3-0  52  20      F         -1 Negative   CTR</span></span>
<span><span class="co">## CCIS15794887ST-4-0  37  18      F         -1 Negative   CTR</span></span>
<span><span class="co">## CCIS74726977ST-3-0  66  24      M         -1 Negative   CTR</span></span>
<span><span class="co">## CCIS16561622ST-4-0  54  26      M         -1 Negative   CTR</span></span>
<span><span class="co">## CCIS79210440ST-3-0  65  30      M         -1 Positive   CTR</span></span>
<span><span class="co">## CCIS82507866ST-3-0  57  24      M         -1 Negative   CTR</span></span></code></pre>
<p>In order to tell <code>SIAMCAT</code>, which samples are cancer cases
and which are healthy controls, we can construct a label object from the
<code>Group</code> column in the metadata.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">label.crc.zeller</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.label.html">create.label</a></span><span class="op">(</span>meta<span class="op">=</span><span class="va">meta.crc.zeller</span>,</span>
<span>    label<span class="op">=</span><span class="st">'Group'</span>, case<span class="op">=</span><span class="st">'CRC'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Label used as case:</span></span>
<span><span class="co">##    CRC</span></span>
<span><span class="co">## Label used as control:</span></span>
<span><span class="co">##    CTR</span></span></code></pre>
<pre><code><span><span class="co">## + finished create.label.from.metadata in    0 s</span></span></code></pre>
<p>Now we have all the ingredients to create a <code>SIAMCAT</code>
object. Please have a look at the <a href="SIAMCAT_read-in.html">vignette about input formats</a> for more
information about supported formats and other ways to create a
<code>SIAMCAT</code> object.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/siamcat.html">siamcat</a></span><span class="op">(</span>feat<span class="op">=</span><span class="va">feat.crc.zeller</span>,</span>
<span>    label<span class="op">=</span><span class="va">label.crc.zeller</span>,</span>
<span>    meta<span class="op">=</span><span class="va">meta.crc.zeller</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## + starting validate.data</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking overlap between labels and features</span></span></code></pre>
<pre><code><span><span class="co">## + Keeping labels of 141 sample(s).</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking sample number per class</span></span></code></pre>
<pre><code><span><span class="co">## +++ checking overlap between samples and metadata</span></span></code></pre>
<pre><code><span><span class="co">## + finished validate.data in 0.234 s</span></span></code></pre>
<p>A few information about the <code>SIAMCAT</code> object can be
accessed with the <code>show</code> function from <code>phyloseq</code>
(<code>SIAMCAT</code> builds on the <code>phyloseq</code> data
structure):</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## siamcat-class object</span></span>
<span><span class="co">## label()                Label object:         88 CTR and 53 CRC samples</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## contains phyloseq-class experiment-level object @phyloseq:</span></span>
<span><span class="co">## phyloseq@otu_table()   OTU Table:            [ 1754 taxa and 141 samples ]</span></span>
<span><span class="co">## phyloseq@sam_data()    Sample Data:          [ 141 samples by 6 sample variables ]</span></span></code></pre>
<p>Since we have quite a lot of microbial species in the dataset at the
moment, we can perform unsupervised feature selection using the function
<code>filter.features</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.features.html">filter.features</a></span><span class="op">(</span><span class="va">sc.obj</span>,</span>
<span>    filter.method <span class="op">=</span> <span class="st">'abundance'</span>,</span>
<span>    cutoff <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Features successfully filtered</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="association-testing">Association Testing<a class="anchor" aria-label="anchor" href="#association-testing"></a>
</h2>
<p>Associations between microbial species and the label can be tested
with the <code>check.associations</code> function. The function computes
for each species the significance using a non-parametric Wilcoxon test
and different effect sizes for the association (e.g. AUC or fold
change).</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/check.associations.html">check.associations</a></span><span class="op">(</span><span class="va">sc.obj</span>, log.n0 <span class="op">=</span> <span class="fl">1e-06</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/association.plot.html">association.plot</a></span><span class="op">(</span><span class="va">sc.obj</span>, sort.by <span class="op">=</span> <span class="st">'fc'</span>, </span>
<span>                panels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'fc'</span>, <span class="st">'prevalence'</span>, <span class="st">'auroc'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The function produces a pdf file as output, since the plot is
optimized for a landscape DIN-A4 layout, but can also used to plot on an
active graphic device, e.g. in RStudio. The resulting plot then looks
like that: <img src="associations_plot.png" alt="Association Plot"></p>
</div>
<div class="section level2">
<h2 id="confounder-testing">Confounder Testing<a class="anchor" aria-label="anchor" href="#confounder-testing"></a>
</h2>
<p>As many biological and technical factors beyond the primary phenotype
of interest can influence microbiome composition, simple association
studies may suffer confounding by other variables, which can lead to
spurious results. The <code>check.confounders</code> function provides
the option to test the associated metadata variables for potential
confounding influence. No information is stored in the
<code>SIAMCAT</code> object, but the different analyses are visualized
and saved to a combined pdf file for qualitative interpretation.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/check.confounders.html">check.confounders</a></span><span class="op">(</span><span class="va">sc.obj</span>, fn.plot <span class="op">=</span> <span class="st">'confounder_plots.pdf'</span>,</span>
<span>                    meta.in <span class="op">=</span> <span class="cn">NULL</span>, feature.type <span class="op">=</span> <span class="st">'filtered'</span><span class="op">)</span></span></code></pre></div>
<p>The conditional entropy check primarily serves to remove nonsensical
variables from subsequent checks. Conditional entropy quantifies the
unique information contained in one variable (row) respective to another
(column). Identical variables and derived variables which share the
exact same information will have a value of zero. In this example, the
label was derived from the Group variable which was determined from AJCC
stage, so both are excluded.</p>
<div class="figure">
<img src="confounder_entropy_plot.png" alt=""><p class="caption">Conditional Entropy Plot</p>
</div>
<p>To better quantify potential confounding effects of metadata
variables on individual microbial features,
<code>check.confounder</code> plots the variance explained by the label
in comparison with the variance explained by the metadata variable for
each individual feature. Variables with many features in the upper left
corner might be confounding the label associations.</p>
<div class="figure">
<img src="confounder_anova_plot.png" alt=""><p class="caption">Variance Explained Plot</p>
</div>
</div>
<div class="section level2">
<h2 id="model-building">Model Building<a class="anchor" aria-label="anchor" href="#model-building"></a>
</h2>
<p>One strength of <code>SIAMCAT</code> is the versatile but easy-to-use
interface for the construction of machine learning models on the basis
of microbial species. <code>SIAMCAT</code> contains functions for data
normalization, splitting the data into cross-validation folds, training
the model, and making predictions based on cross-validation instances
and the trained models.</p>
<div class="section level3">
<h3 id="data-normalization">Data Normalization<a class="anchor" aria-label="anchor" href="#data-normalization"></a>
</h3>
<p>Data normalization is performed with the
<code>normalize.features</code> function. Here, we use the
<code>log.unit</code> method, but several other methods and
customization options are available (please check the
documentation).</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.features.html">normalize.features</a></span><span class="op">(</span><span class="va">sc.obj</span>, norm.method <span class="op">=</span> <span class="st">"log.unit"</span>,</span>
<span>    norm.param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>log.n0 <span class="op">=</span> <span class="fl">1e-06</span>, n.p <span class="op">=</span> <span class="fl">2</span>,norm.margin <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Features normalized successfully.</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="prepare-cross-validation">Prepare Cross-Validation<a class="anchor" aria-label="anchor" href="#prepare-cross-validation"></a>
</h3>
<p>Preparation of the cross-validation fold is a crucial step in machine
learning. <code>SIAMCAT</code> greatly simplifies the set-up of
cross-validation schemes, including stratification of samples or keeping
samples inseperable based on metadata. For this small example, we choose
a twice-repeated 5-fold cross-validation scheme. The data-split will be
saved in the <code>data_split</code> slot of the <code>SIAMCAT</code>
object.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/create.data.split.html">create.data.split</a></span><span class="op">(</span><span class="va">sc.obj</span>, num.folds <span class="op">=</span> <span class="fl">5</span>, num.resample <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Features splitted for cross-validation successfully.</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="model-training">Model Training<a class="anchor" aria-label="anchor" href="#model-training"></a>
</h3>
<p>The actual model training is performed using the function
<code>train.model</code>. Again, multiple options for customization are
available, ranging from the machine learning method to the measure for
model selection or customizable parameter set for hyperparameter
tuning.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/train.model.html">train.model</a></span><span class="op">(</span><span class="va">sc.obj</span>, method <span class="op">=</span> <span class="st">"lasso"</span><span class="op">)</span></span></code></pre></div>
<p>The models are saved in the <code>model_list</code> slot of the
<code>SIAMCAT</code> object. The model building is performed using the
<code>mlr</code> R package. All models can easily be accessed.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get information about the model type</span></span>
<span><span class="fu"><a href="../reference/model_type-methods.html">model_type</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "lasso"</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># access the models</span></span>
<span><span class="va">models</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/models-methods.html">models</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span>
<span><span class="va">models</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">model</span></span></code></pre></div>
<pre><code><span><span class="co">## &lt;LearnerClassifCVGlmnet:classif.cv_glmnet&gt;</span></span>
<span><span class="co">## * Model: cv.glmnet</span></span>
<span><span class="co">## * Parameters: alpha=1, s=0.02981</span></span>
<span><span class="co">## * Packages: mlr3, mlr3learners, glmnet</span></span>
<span><span class="co">## * Predict Types:  response, [prob]</span></span>
<span><span class="co">## * Feature Types: logical, integer, numeric</span></span>
<span><span class="co">## * Properties: multiclass, selected_features, twoclass, weights</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="make-predictions">Make Predictions<a class="anchor" aria-label="anchor" href="#make-predictions"></a>
</h3>
<p>Using the data-split and the models trained in previous step, we can
use the function <code>make.predictions</code> in order to apply the
models on the test instances in the data-split. The predictions will be
saved in the <code>pred_matrix</code> slot of the <code>SIAMCAT</code>
object.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make.predictions.html">make.predictions</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span>
<span><span class="va">pred_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pred_matrix-methods.html">pred_matrix</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pred_matrix</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                       CV_rep1    CV_rep2</span></span>
<span><span class="co">## CCIS27304052ST-3-0 0.09197911 0.12133887</span></span>
<span><span class="co">## CCIS15794887ST-4-0 0.15167414 0.08921955</span></span>
<span><span class="co">## CCIS74726977ST-3-0 0.34401535 0.32544187</span></span>
<span><span class="co">## CCIS16561622ST-4-0 0.21979245 0.19206815</span></span>
<span><span class="co">## CCIS79210440ST-3-0 0.33730175 0.06324943</span></span>
<span><span class="co">## CCIS82507866ST-3-0 0.15877907 0.07749619</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="model-evaluation-and-interpretation">Model Evaluation and Interpretation<a class="anchor" aria-label="anchor" href="#model-evaluation-and-interpretation"></a>
</h2>
<p>In the final part, we want to find out how well the model performed
and which microbial species had been selected in the model. In order to
do so, we first calculate how well the predictions fit the real data
using the function <code>evaluate.predictions</code>. This function
calculates the Area Under the Receiver Operating Characteristic (ROC)
Curve (AU-ROC) and the Precision Recall (PR) Curve for each resampled
cross-validation run.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sc.obj</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Evaluated predictions successfully.</span></span></code></pre>
<div class="section level3">
<h3 id="evaluation-plot">Evaluation Plot<a class="anchor" aria-label="anchor" href="#evaluation-plot"></a>
</h3>
<p>To plot the results of the evaluation, we can use the function
<code>model.evaluation.plot</code>, which produces a pdf-file showing
the ROC and PR Curves for the different resamples runs as well as the
mean ROC and PR Curve.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/model.evaluation.plot.html">model.evaluation.plot</a></span><span class="op">(</span><span class="va">sc.obj</span><span class="op">)</span></span></code></pre></div>
<p><img src="eval_plot.png"></p>
</div>
<div class="section level3">
<h3 id="interpretation-plot">Interpretation Plot<a class="anchor" aria-label="anchor" href="#interpretation-plot"></a>
</h3>
<p>The final plot produced by <code>SIAMCAT</code> is the model
interpretation plot, created by the
<code>model.interpretation.plot</code> function. The plot shows for the
top selected features the</p>
<ul>
<li><p>model weights (and how robust they are) as a barplot,</p></li>
<li><p>a heatmap with the z-scores or fold changes for the top selected
features, and</p></li>
<li><p>a boxplot showing the proportions of weight per model which is
captured by the top selected features.</p></li>
</ul>
<p>Additionally, the distribution of metadata is shown in a heatmap
below.</p>
<p>The function again produces a pdf-file optimized for a landscape
DIN-A4 plotting region.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/model.interpretation.plot.html">model.interpretation.plot</a></span><span class="op">(</span><span class="va">sc.obj</span>, fn.plot <span class="op">=</span> <span class="st">'interpretation.pdf'</span>,</span>
<span>    consens.thres <span class="op">=</span> <span class="fl">0.5</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span>, heatmap.type <span class="op">=</span> <span class="st">'zscore'</span><span class="op">)</span></span></code></pre></div>
<p>The resulting plot looks like this: <img src="interpretation_plot.png" alt="Model Interpretation Plot"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] SIAMCAT_2.3.3    phyloseq_1.42.0  mlr3_0.14.1      BiocStyle_2.26.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] paradox_0.11.0         minqa_1.2.5            colorspace_2.1-0      </span></span>
<span><span class="co">##   [4] ellipsis_0.3.2         rprojroot_2.0.3        XVector_0.38.0        </span></span>
<span><span class="co">##   [7] fs_1.6.1               rstudioapi_0.14        listenv_0.9.0         </span></span>
<span><span class="co">##  [10] mlr3tuning_0.18.0      fansi_1.0.4            codetools_0.2-19      </span></span>
<span><span class="co">##  [13] splines_4.2.2          mlr3learners_0.5.6     PRROC_1.3.1           </span></span>
<span><span class="co">##  [16] cachem_1.0.7           knitr_1.42             ade4_1.7-22           </span></span>
<span><span class="co">##  [19] jsonlite_1.8.4         nloptr_2.0.3           pROC_1.18.0           </span></span>
<span><span class="co">##  [22] gridBase_0.4-7         cluster_2.1.4          BiocManager_1.30.20   </span></span>
<span><span class="co">##  [25] compiler_4.2.2         backports_1.4.1        Matrix_1.5-3          </span></span>
<span><span class="co">##  [28] fastmap_1.1.1          cli_3.6.0              prettyunits_1.1.1     </span></span>
<span><span class="co">##  [31] htmltools_0.5.4        tools_4.2.2            lmerTest_3.1-3        </span></span>
<span><span class="co">##  [34] igraph_1.4.1           gtable_0.3.1           glue_1.6.2            </span></span>
<span><span class="co">##  [37] GenomeInfoDbData_1.2.9 reshape2_1.4.4         LiblineaR_2.10-22     </span></span>
<span><span class="co">##  [40] dplyr_1.1.0            Rcpp_1.0.10            Biobase_2.58.0        </span></span>
<span><span class="co">##  [43] jquerylib_0.1.4        pkgdown_2.0.7          vctrs_0.5.2           </span></span>
<span><span class="co">##  [46] Biostrings_2.66.0      rhdf5filters_1.10.0    multtest_2.54.0       </span></span>
<span><span class="co">##  [49] ape_5.7-1              nlme_3.1-162           iterators_1.0.14      </span></span>
<span><span class="co">##  [52] xfun_0.37              stringr_1.5.0          mlr3measures_0.5.0    </span></span>
<span><span class="co">##  [55] globals_0.16.2         lme4_1.1-32            lifecycle_1.0.3       </span></span>
<span><span class="co">##  [58] beanplot_1.3.1         future_1.32.0          zlibbioc_1.44.0       </span></span>
<span><span class="co">##  [61] MASS_7.3-58.3          scales_1.2.1           lgr_0.4.4             </span></span>
<span><span class="co">##  [64] hms_1.1.2              ragg_1.2.5             parallel_4.2.2        </span></span>
<span><span class="co">##  [67] biomformat_1.26.0      rhdf5_2.42.0           RColorBrewer_1.1-3    </span></span>
<span><span class="co">##  [70] yaml_2.3.7             memoise_2.0.1          gridExtra_2.3         </span></span>
<span><span class="co">##  [73] ggplot2_3.4.1          sass_0.4.5             stringi_1.7.12        </span></span>
<span><span class="co">##  [76] S4Vectors_0.36.2       desc_1.4.2             corrplot_0.92         </span></span>
<span><span class="co">##  [79] foreach_1.5.2          checkmate_2.1.0        permute_0.9-7         </span></span>
<span><span class="co">##  [82] palmerpenguins_0.1.1   BiocGenerics_0.44.0    boot_1.3-28.1         </span></span>
<span><span class="co">##  [85] shape_1.4.6            GenomeInfoDb_1.34.9    matrixStats_0.63.0    </span></span>
<span><span class="co">##  [88] rlang_1.1.0            pkgconfig_2.0.3        systemfonts_1.0.4     </span></span>
<span><span class="co">##  [91] bitops_1.0-7           evaluate_0.20          lattice_0.20-45       </span></span>
<span><span class="co">##  [94] purrr_1.0.1            Rhdf5lib_1.20.0        tidyselect_1.2.0      </span></span>
<span><span class="co">##  [97] parallelly_1.34.0      plyr_1.8.8             magrittr_2.0.3        </span></span>
<span><span class="co">## [100] bookdown_0.33          R6_2.5.1               IRanges_2.32.0        </span></span>
<span><span class="co">## [103] generics_0.1.3         DBI_1.1.3              pillar_1.8.1          </span></span>
<span><span class="co">## [106] mgcv_1.8-42            survival_3.5-5         RCurl_1.98-1.10       </span></span>
<span><span class="co">## [109] tibble_3.2.0           crayon_1.5.2           uuid_1.1-0            </span></span>
<span><span class="co">## [112] utf8_1.2.3             rmarkdown_2.20         progress_1.2.2        </span></span>
<span><span class="co">## [115] grid_4.2.2             data.table_1.14.8      vegan_2.6-4           </span></span>
<span><span class="co">## [118] infotheo_1.2.0.1       mlr3misc_0.11.0        bbotk_0.7.2           </span></span>
<span><span class="co">## [121] digest_0.6.31          numDeriv_2016.8-1.1    textshaping_0.3.6     </span></span>
<span><span class="co">## [124] stats4_4.2.2           munsell_0.5.0          glmnet_4.1-6          </span></span>
<span><span class="co">## [127] bslib_0.4.2</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Konrad Zych, Jakob Wirbel, Georg Zeller.</p>
</div>

<div class="privacy">
  <p><strong>Privacy:</strong> Usage of this site follows
      <a href="https://www.embl.de/aboutus/privacy_policy/index.html" class="external-link">EMBL’s Privacy Policy</a><br>
      In accordance with that policy, we use Matomo to collect anonymised data on visits to, downloads from, and searches of this site.
      Contact <a href="mailto:zeller@embl.de">Georg Zeller</a> for details.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
